---
title: 'TADCompare: an R package for differential analysis and characterization of Topologically Associated Domains'
fontsize: 12pt
mainfont: Times New Roman
csl: styles.ref/genome-research.csl
header-includes:
   - \usepackage{float}
   - \usepackage{setspace}\doublespacing
output:
  word_document:
    reference_docx: styles.doc/Arial_11_single_space_normal_margins.docx
  pdf_document:
    fig_caption: no
  html_document: null
bibliography: [3D_refs.bib,tad_references.bib,difftad_references.bib]
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='figures/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
library(dplyr)
options(stringsAsFactors = FALSE)
```


Kellen G. Cresswell^1,2^ (cresswellkg@vcu.edu), Mikhail G. Dozmorov^1,2\*^ (mikhail.dozmorov@vcuhealth.org)

^1^ Dept. of Biostatistics, Virginia Commonwealth University, Richmond, VA, 23298, USA  
^2^ Equal contribution  
^\*^ To whom correspondence should be addressed: Virginia Commonwealth University, Richmond, VA, 23298, 804-827-2055, mikhail.dozmorov@vcuhealth.org

**Keywords:** Hi-C, chromosome conformation capture, topologically associated domains, TADs, differential analysis, TADCompare

**Running Title:** TADCompare: an R package for differential TAD detection

# Abstract

Recent research using chromatin conformation capture technologies, such as Hi-C, has demonstrated the importance of topologically associated domains (TADs). TADs are considered to be relatively stable three-dimensional (3D) genomic structures orchestrating gene expression regulation; yet, many dynamically reorganize during development or disease, and exhibit cell- and condition-specific differences. Quantification of the dynamic behavior of TADs will help to better understand genome regulation. Methods for comparing TADs between cells and conditions are highly limited. We developed `TADCompare`, a method for differential analysis of TAD boundaries between Hi-C datasets. `TADCompare` is based on a spectral clustering-derived measure called the eigenvector gap, which enables a loci-by-loci comparison of TAD boundary differences between datasets. Using this measure, we introduce methods for identifying differential and consensus TAD boundaries, and tracking TAD boundary changes over time. We further propose a novel framework for the systematic classification of TAD boundary changes. Colocalization- and gene enrichment analysis of different types of TAD boundary changes demonstrated distinct biological functionality associated with them. `TADCompare` is available on https://github.com/dozmorovlab/TADCompare.

### Keywords

# Introduction

Recent research indisputably proves importance of the 3D genome organization in regulating gene expression and other genomic processes [@Schoenfelder:2010ab; @Steensel:2011aa; @Franke:2016aa; @Symmons:2014aa; @Sexton:2015aa; @Li:2012aa; @Papantonis:2013aa; @Laat:2003aa; @Mora:2016aa; @Mifsud:2015aa; @Shavit:2014aa; @Osborne:2004aa; @Schoenfelder:2010aa; @Tanizawa:2010aa]. The 3D genomic structures consists of chromosome territories [@Cremer:2010aa], A/B compartments corresponding to active/repressed chromatin [@Rao:2014aa; @Lieberman-Aiden:2009aa], topologically associated domains (TADs) [@Dixon:2012aa; @Jackson:1998aa; @Ma:1998aa; @Nora:2012aa; @Sexton:2012aa; @Bonev:2017aa; @Dekker:2002aa], smaller sub-TADs [@Phillips-Cremins:2013aa; @Rao:2014aa] and chromatin loops [@Rao:2014aa; @Dowen:2014aa; @Ji:2016aa; @Denker:2016aa]. These structures help to regulate global gene expression [@Schoenfelder:2010ab; @Steensel:2011aa; @Franke:2016aa; @Symmons:2014aa; @Sexton:2015aa; @Li:2012aa; @Papantonis:2013aa; @Laat:2003aa; @Mora:2016aa; @Mifsud:2015aa; @Shavit:2014aa; @Osborne:2004aa; @Schoenfelder:2010aa; @Tanizawa:2010aa]. Consequently, coordinated changes in the 3D structures [@Yaffe:2011aa; @Symmons:2014aa; @Dai:2012aa] determine cell type-specific gene expression and identity [@Dowen:2014aa; @Ji:2016aa; @Phillips-Cremins:2013aa; @Rao:2014aa; @Vietri-Rudan:2015aa; @Jin:2013aa; @Schoenfelder:2010aa; @Dekker:2013aa], guide recombination [@Jhunjhunwala:2009aa], X chromosome inactivation [@Nora:2012aa; @Crane:2015aa]. Many 3D structures are largely invariant between different cell types, and even conserved between mammalian species [@Dixon:2012aa; @Nora:2012aa; @Pope:2014aa; @Naumova:2013aa; @Rao:2014aa; @Vietri-Rudan:2015aa], indicating their high biological importance during genome evolution.

Despite the high level of conservation, recent research uncovered the dynamic nature of the 3D genomic structures, and this plasticity accompanies various biological functions and phenomena [@Yu:2017aa]. In Drosophila, exposure to heat-shock caused local changes in certain TAD boundaries resulting in TAD merging [@Li:2015ac]. Another recent study showed that during motor neuron (MN) differentiation in mammals, TAD, and sub-TAD boundaries in the Hox cluster are not rigid and their plasticity is linked to changes in gene expression during differentiation [@Narendra:2016aa]. The global organization of the 3D genomic structure is found in mitosis [@Nagano:2017aa], <!--zygotic development [@Hug:2017aa; @Flyamer:2017aa], -->earliest stages of mammalian lineage development [@Dixon:2015aa; @Du:2017aa; @Ke:2017aa; @Bonev:2017aa], and somatic cell reprogramming of pluripotent stem cells [@Zhang:2018aa; @Novo:2018aa]. Fusion of TADs [@Flavahan:2016aa; @Fudenberg:2016aa; @Sanborn:2015aa; @Dowen:2014aa; @Guo:2015aa; @Tang:2015aa; @Nora:2012aa], creation or destruction of sub-TADs within existing TAD boundaries [@Taberlay:2016aa; @Lupianez:2016aa], and/or switching TAD states between active and inactive conformations [@Dixon:2012aa; @Lieberman-Aiden:2009aa] has been associated with a variety of phenotypes [@Krijger:2016aa; @Misteli:2010aa; @Spielmann:2018aa], ranging from limb malformation [@Lupianez:2016aa], congenital disorders [@Ibn-Salem:2014aa], to cancer [@Mitelman:2000aa; @Valton:2016aa; @Rickman:2012aa; @Hnisz:2016aa; @Groschel:2014aa; @Corces:2016aa; @Lupianez:2016aa; @Krijger:2016aa; @Flavahan:2016aa; @Barutcu:2015aa]. These observations highlight the importance of studying changes in TADs as a means to understand genomic regulation. However, methods for identifying changes in TAD boundaries remain underdeveloped.

To our knowledge, there are only three methods that can be adapted for detecting changes in TAD boundaries: `localtadsim` [@Sauerwald:2018aa2], `HiCDB` [@Chen:2018aa] and `DiffTAD` [@Zaborowski:2016aa]. Of the three methods, none provide an intuitive, easy to use way of calling differential TADs. Both `localtadsim` and `DiffTAD` are two-step procedures requiring separately defined TADs and comparing them using a command-line utility. `HiCDB` has a built-in TAD caller but does not allow for comparisons of chromosome-specific contact matrices. All three methods require highly specific data types and file names to be able to run. The lack of usability is compounded with issues such as a lack of upkeep, slow runtimes, and lack of statistical rigor (Supplementary methods).

As costs of Hi-C data continue to drop, several studies started to investigate the dynamics of 3D changes over time. The most notable applications include cell differentiation studies [@Bonev:2017aa], embryonic development [@Du:2017aa;@Hug:2017aa;@Ke:2017aa], cancer progression [@Zhou:2019aa]. Typically, TAD boundary changes over time are quantified by overlap [@Hug:2017aa;@Du:2017aa] and classified into distinct patterns [@Zhou:2019aa]. However, general-purpose methods for systematic analysis of TAD boundary changes over time do not exist. Furthermore, the number of replicates for a given experiment continues to rise, requiring methods for defining and comparing TAD boundaries across replicates of Hi-C data. <!--Previous research has shown inconsistencies between both technological and biological replicates [@Dixon:2012aa;@Rao:2014aa;@Sauerwald:2018aa2]. As a result, researchers must take this variation into account and make full use of the data. -->Traditionally, two approaches have been developed to identify TAD boundaries across replicates. The first is to call TADs on individual replicates and aggregate them (e.g., `Arrowhead` [@Rao:2014aa]). The second approach involves combining all replicates into a consensus contact matrix <!--using utilities such as `Juicer` [@Durand:2016aa] -->and then calling TADs (`Arboretum` [@Fotuhi-Siahpirani:2016aa]). [??? TBD - are the tools mentioned to the corresponding methods?] However, it is focused on identifying TADs across cell-lines and species as opposed to across replicates. Methods for detecting consensus TADs across Hi-C datasets remain underdeveloped.

We developed `TADCompare`, an R package aimed at providing a fast, accurate, user-friendly and well-documented approach to differential TAD analysis. We introduce a method based on the boundary score statistic [@Cresswell:2019aa] and use it to identify five types of TAD boundary changes. The method is extended to allow for calling consensus TAD boundaries and comparing them between groups of Hi-C replicates. We further demonstrate how the TAD boundary score statistic may be used to analyze TAD dynamics over the time course. For both differential TAD boundary detection and time course analysis, we provide novel terminology for the classification of TAD boundary changes. We demonstrated the robustness of `TADCompare` using simulated data with pre-defined TADs [@Forcato:2017aa] and its ability to reveal distinct biological roles of different TAD boundary changes. In summary, `TADCompare` provides an all-in-one pipeline from TAD calling to differential boundary detection, including time course, that supports replicated Hi-C data analysis. The output is formatted in a commonly used BED format that allows for flexible downstream analyses and visualization. The `TADCompare` R package is freely available on GitHub (https://github.com/dozmorovlab/TADCompare). 

# Methods

### Representation of Hi-C data as a graph

For a given Hi-C experiment, Hi-C data is represented by a chromosome-specific contact matrix $C$ of non-overlapping regions (aka bins) of size $r$ (resolution of the data). Each entry $C_{ij}$ corresponds to the number of contacts between region $i$ and region $j$. Previous work has shown that this contact matrix is essentially an analog of the adjacency matrix found in graph theory and Hi-C data can be thought of as a naturally occurring graph where edges are contacts and vertices are genomic regions [@Boulos:2013aa; @wang2017network; @Wang:2013:TPC:2506583.2506633;@Cresswell:2019aa]. The graph representation of Hi-C data is the foundation of our method and allows us to use a graph-clustering based approach to identify and analyze TADs.

### Calculating the graph spectrum

The first step of our method is to calculate the graph spectrum, defined as the eigenvectors of the Laplacian of an adjacency matrix. Using the interpretation of the contact matrix as a naturally occurring adjacency matrix, we calculate the Laplacian directly from the contact data. Briefly, the graph spectrum for a given contact matrix is calculated as follows:

1. Calculate the normalized Laplacian $\bar{L}$:

$$
\bar{L} = D^{-\frac{1}{2}}CD^{-\frac{1}{2}}
$$

where $D = diag(\textbf{1}^{T}C)$, where $\textbf{1}$ is a column vector of size $C$ where each entry is 1. $D$ can be thought of as a vector containing the sum of the degrees for a given node. 

2. Perform an eigendecompositon of the Laplacian:

$$
\bar{L}v = \lambda v
$$

In practice, we calculate the first two eigenvectors with the largest absolute values of eigenvalues and organize them into a matrix $\bar{V}$ with dimensions $i \times 2$, where $i$ is the number of regions in the contact matrix. $\bar{V}$ is referred to as the graph spectrum of the contact matrix.

### Eigenvector gap as a measure of pattern change

We can think of each row of the matrix $\bar{V}$ as a quantification of the pattern of contacts in each region of the contact matrix. Previous work [@Cresswell:2019aa] has demonstrated that by taking the Euclidean distance between row $V_{i.}$ and its neighboring row $V_{(i+1).}$, one can measure the similarity in the pattern of contacts between region $i$ and region $i+1$ of the chromosome, termed "eigenvector gap". A TAD boundary manifests itself as a sudden break in the pattern of contacts. This pattern is reflected in the eigenvector gap by a spike in gap size followed by and preceded by smaller gaps (Figure 1). The eigenvector gap quantifies the degree of this break, acting as a proxy for TAD boundary likelihood. To calculate the eigenvector gaps, we perform the following procedure:

1. Normalize columns of $\bar{V}$ to sum to 1:

$$
\begin{aligned}
\hat{V_{ij}} = \frac{\bar{V_{ij}}}{ \Vert{\bar{V_{.j}\Vert}}}
\end{aligned}
$$

where the subscript $.j$ corresponds to column $j$.

2. Normalize $\hat{V}$ and project onto a unit circle:

$$
\tilde{Z} = Diag(diag^{-\frac{1}{2}}(\hat{V_{i.}}\hat{V_{i.}}^{T}))\hat{V_{i.}}
$$
<!--$\tilde{Z}$ is the projection of $\bar{V}$ onto a unit circle such that all values are bounded by [-1,1].-->

3. Calculate the distance between neighboring regions (rows $i$ and $i-1$ of $\tilde{Z}$) and store in a vector $D_{i}$:

$$
D_{i}=\sqrt{(\tilde{Z}_{i1}-\tilde{Z}_{(i-1)1})^2 + (\tilde{Z}_{i2}-\tilde{Z}_{(i-1)2})^2}
$$ 

We refer to $D$ as the vector where each entry $D_i$ is referred to as an eigenvector gap. Formally, an eigenvector gap is the Euclidean distance between each successive row of the first two eigenvectors. In practical terms, the eigenvector gap for a given locus is a measure of how likely that locus is a TAD boundary. 

To maintain the association of each entry of the vector with its corresponding matrix region, a placeholder is used in the first entry of the vector. This is necessary because we cannot calculate an eigenvector gap for the first entry of the contact matrix due to a lack of a left-bound neighbor. In mathematical terms, this means that for a matrix of size $n$ the total number of eigenvector gaps is $n-1$.

<!--
Graph spectrum calculation scales poorly for large matrices, having a quadratic complexity [@Wu:2018aa]. By using a windowed approach, the complexity becomes linear, relying mainly on the window size to determine the speed of the algorithm. The window starts at the beginning of the contact matrix. After calculating the eigenvector gaps within the window, the window moves forward $w-1$ bins across the diagonal where $w$ is the size of the window. We move $w-1$ spaces because for a given window it is impossible to calculate the eigenvector gap of the last bin. Moving $w-1$ allows us to include that last bin in the next window and calculate this value. We continue this process until the end of the contact matrix is reached.  
-->

### Converting eigenvector gaps to boundary scores

We showed that the distribution of eigenvector gaps can be approximated by a log-normal distribution (Supplementary Figure 1, Supplementary Material). The log-normality allows us to convert the eigenvector gap values into boundary scores:

$$
B_{i} = \frac{ (ln(D_i) - \mu)} {\sigma^{2}}
$$

where $ln(D) \sim N(\mu, \sigma^{2})$ where $\mu$ and $\sigma^{2}$ are the mean and variance of the distribution of the natural log of the eigenvector gaps, respectively, and $B$ is a vector of boundary scores with a N(0,1) distribution. In practice, this value is simply the Z-score for the natural log of eigenvector gaps.

### Sliding window eigenvector gap calculation

Frequency of interactions decays following power law as the distance between the interacting regions in a linear genome increases [@Lajoie:2015aa]. This decay leads to noisy and non-informative interactions farther off-diagonal of the contact matrix. To alleviate the effect of noisy distant interactions, we perform spectral decomposition within a fixed-size window that moves along the diagonal of the matrix. For instance, a window size of 15 bins (default setting, Supplementary Figure 2, Supplementary Material) means that only values within 15 bins of the diagonal will be used to calculate the eigenvector gap. The sliding window approach improves the performance of eigenvector gap calculation [@Cresswell:2019aa]. Additionally, it provides for faster calculations, operating on many small matrices instead of one large matrix.

<!-- ### Measuring contact matrix similarity using boundary scores -->

<!-- Oftentimes, it is of interest to measure the similarity of entire contact matrices. This is useful for comparing technical or biological replicates and assessing the consistency of Hi-C sequencing. Calculating similarity is complicated by the need to capture the topological similarity of contact matrices and certain issues such as distance decay and noise. Previous methods have attempted to quantify this similarity, such as Quasar-Rep [@Sauria:2015aa], Hi-C-Spector[@Yardimci:2017aa], Hi-CRep [@Yang:2017aa] and GenomeDISCO [@Ursu:2018aa]. Both the Quasar-Rep and approach takes two contact matrices, transforms them by stratifying based on distance, and then takes a Pearson correlation. GenomeDISCO and Hi-C-Spector are graph-based methods, with GenomeDISCO measuring similarity based on the distance between graph-diffusion smoothed contact matrices and Hi-C-Spector measuring the distance between eigenvectors of the Laplacians of each contact matrix. We propose a method based on cross-correlation of boundaries to calculate similarity. -->

<!-- Given two contact matrices $P$ and $R$ with boundary scores of $B_P$ and $B_R$ having means of $\mu_{P}$ and $\mu_{R}$ respectively, we can calculate matrix similarity using a modified form of cross-correlation that takes into account small shifts in TAD boundaries. For this example, we first define the basic cross-correlation of two sets of boundary scores with lag $l$: -->

<!-- $$ -->
<!-- S = \frac{ \sum[(B_{P}(i)-\mu_{P})*(B_{R}(i-l)-\mu_{R})] -->

<!-- }{\sqrt{\sum(B_{P}(i)-\mu_{P})^{2}}\sqrt{\sum(B_{R}(i-l)-\mu_{R})^{2}}} -->
<!-- $$ -->

<!-- where $i$ is the corresponding vector entry and $l$ is the lag. Using this framework, we incorporate the fact that shifts of less than 5 are biologically insignificant by replacing all instances of $B_{R}(i-l)$ with $B_{min}(i) = min(|(B_{P}(i)-\mu_{P})-(B_{R}(i-l)-\mu_{R})|)$ for l = 1,...,5. The resulting formula is the following: -->

<!-- $$ -->
<!-- S_{adj} = \frac{ \sum_{i}[(B_{P}(i)-\mu_{P})*(B_{min}(i)-\mu_{R})] -->

<!-- }{\sqrt{\sum_{i}(B_{P}(i)-\mu_{P})^{2}}\sqrt{\sum_{i}(B_{min}(i)-\mu_{R})^{2}}} -->
<!-- $$ -->

<!-- $S_{adj}$ gives us a measure of similarity between contact matrices that gives equal weighting to lagged and non-lagged TAD boundaries, finding the region of maximum similarity. This score effectively provides a measure of similarity between contact matrices purely in terms of TAD similarity. The adjustment allows us to treat shifted boundaries the same as non-shifted boundaries and more accurately capture contact matrix similarity.   -->

### Differential analysis using boundary scores

To define the differences between two contact matrices, $P$ and $R$, we compare their eigenvector gaps $D_P$ and $D_R$, respectively. Given that $ln(D_P) \sim N(\mu_{P}, \sigma_{P}^{2})$ and $ln(D_R) \sim N(\mu_{R}, \sigma_{R}^{2})$, it follows that $ln(D_P) - ln(D_R) \sim N(\mu_{P}-\mu_{R}, \sigma_{P}^{2}+\sigma_{R}^{2})$. These results allow us to calculate a vector of differential boundary scores:

$$
DB_{i} = \frac{ (ln(D_{Pi}) - ln(D_{Ri})) - (\mu_{P}-\mu_{R}) } {\sigma_{P}^{2}+\sigma_{R}^{2}}
$$

or more simply,

$$
DB_{i} = \frac{\sigma_{P}^{2}B_{P}-\sigma_{R}^{2}B_{R}}{\sigma_{P}^{2}+\sigma_{R}^{2}}
$$

where $B_{P}$ and $B_{R}$ are the boundary scores for the $P$ and $R$ matrices, respectively. This score can be thought of as the difference in TAD boundary likelihood for a given locus in two data sets. Due to the aforementioned normality of the difference in log eigenvector gaps, $DB_i$ can be thought of as a simple z-score where $DB \sim N(0,1)$.

### Time course boundary changes

Boundary scores provide a convenient method for modeling the change of TAD boundaries over time. For a given TAD boundary, or, any region of the genome, we can monitor the trajectory of the boundary score. Over time, we can define boundary score changes based on their deviation from a baseline level (typically, the first time point). It is expected that these scores will be relatively constant over time except in regions where a boundary appears or disappears. The trend across time points can be recorded, and the pattern of change classified accordingly. Our implementation of time course boundary analysis allows for the usage of multiple replicates for a given time point. Briefly, at each region of the genome, the consensus boundary score is calculated, defined as the median of consensus scores across all replicates, and is then used to identify boundaries. 

<!--The overall trend can be calculated by taking the full set of TAD boundaries across every time point and calculating the average boundary score for each individual point. If this score is increasing, then it suggests the appearance of TAD boundaries over time. Conversely, if it is decreasing, then there is a disappearance of TAD boundaries. We can also incorporate the consensus boundary score framework into boundary score calculations over time. Given a set of replicates at multiple different time points, we can call consensus boundary scores at for each time points and aggregate them. We can also quantify the change in TAD strength for the entire contact matrix over time by modeling the trend of mean boundary scores across each of the time point. A positive trend indicates an increase in TAD strength or abundance and a decrease indicates a drop.-->

<!--
### Simulated data for differential TAD boundary detection

To simulate differential TAD boundaries, we used 25 simulated contact matrices with manually annotated TADs downloaded from Hi-CToolsCompare [@Forcato:2017aa]. These matrices were generated at five levels of noise (4%, 8%, 12%, 16%, and 20%) with five simulated matrices per noise level. Noise is generated by randomly selecting entries in the contact matrix and adding two contacts. The percentage value corresponds to the percentage of matrix entries with constant noise added.

Within each noise level, we take each possible combination of contact matrices and calculate differential TADs between them. It is expected that an accurate differential boundary detection method will find all manually annotated TADs which differ between each combination of simulated matrices. For each combination of simulated matrices, we define true differential TAD boundaries as the set of non-shared boundaries between the matrices. To assess the ability of our method to discriminate between true differential TADs and non-differential TADs, we use the Matthews correlation coefficient. [??? TBD, rationale for MCC]
-->
<!-- ### Primary and replicate experimental data for differential TAD boundary detection -->

<!-- Primary and replicate data from [@Rao:2014aa] was used to test the percentage of differential TAD boundaries between primary and replicate data. TAD boundaries were called using the `SpectralTAD` TAD caller [@Cresswell:2019aa]. The raw percentage of shared boundaries at the first level of the hierarchy were calculated between the primary and replicate. To test whether these boundaries were truly differential, we then ran `TADCompare` and calculated the percentage of non-shared boundaries which were also detected by TADCompare.  -->

<!-- We hypothesized that non-shared boundaries between primary and replicates are largely the result of differences in the TAD hierarchy. If true, it is expected that TAD boundaries which are not shared between primary and replicate and are not detected by `TADCompare` will actually appear somewhere in both the primary and replicate's hierarchy of TADs. [??? Shall we forgo hierarchy?] -->
<!--
### Handling of gaps

Gaps refer to loci with either complete absence of loci or levels of sparsity which make analysis impossible. For our analysis, we defined gaps as any region with less than 80\% of its contacts greater than zero. Since we are not particularly interested in long-range contacts, this percentage is calculated based on regions within our sliding window. These gaps can introduce instability in the algorithm and lack important information. To counter this, we simply remove these regions before the analysis. This is done for both contact matrices such that, if one contact matrix contains a gap at a given locus and the other does not, we remove it from both. This allows us to still make a one-to-one comparison of regions. 
-->

### Data sources

All simulated data were downloaded from `HiCToolsCompare` repository [@Forcato:2017aa]. In total, we used 25 simulated matrices with varying levels of noise. For sparsity and downsampling analysis matrices were manually created based on matrices from `HiCToolsCompare` matrices with the minimum noise level (see [@Cresswell:2019aa] for methods description). Data for comparisons across cell lines, replicates, and tissues were taken from [@Schmitt:2016aa], generated at 40kb resolution (Supplementary Table 1). Time course data was taken from [@Rao:2017aa], HCT-116 human colon cancer cell-line at four time points after auxin-treatment withdrawal (20, 40, 60, 180 minutes). Contact matrices were generated at 25kb, 50kb, and 100kb using the `straw` tool from `Juicer` [@Durand:2016aa]. Chromatin state data were taken from chromHMM [@Ernst:2010aa]. Histone modifications and transcription factor binding sites were downloaded from the Encyclopedia of DNA Elements (ENCODE) [@Davis:2018aa] (Supplementary Table 2).  

### Gene enrichment testing 

All gene enrichment testing was performed using the `rGREAT` (Version 2.0) R package. Briefly, we detect genes within 5kb upstream and 1kb downstream of each type of TAD boundary changes, similar to work of others [@Chen:2018aa]. For each Gene Ontology (GO) and pathway (MSigDb [??? Describe version, add ref]), a hypergeometric test is then performed to determine over-representation of TAD boundary-associated genes. For all figures, we report results for GO Biological Processes. Results for GO Molecular Function, GO Cellular Component, MSigDB and PANTHER pathways are reported in tables.  

### Colocalization enrichment testing

A permutation test was used to quantify the enrichment of colocalization of TAD boundaries of interest with genomic annotations. Briefly, we flank each type of TAD boundary changes (differential or time course) by 50kb on each side and calculate the mean number of genomic annotations across those regions (observed enrichment). Next, we generate two sets of bins, one the size of the TAD boundaries which we are testing (considering the flanking) and another the size of all other bins. The difference in the mean number of genomic annotations colocalized with TAD boundaries of interest was calculated for each set (expected enrichment). We repeat this procedure 10,000 times. We calculate the permutation p-value by taking the number of times the expected enrichment was greater than the observed enrichment, and dividing by 10000. $\alpha = 0.05$ was set to assess statistical significance. 

# Results

### A modified spectral clustering approach is better suited for TAD boundary detection than other approaches

Our previous work on TAD detection using spectral clustering, implemented as a `SpectralTAD` R package [@Cresswell:2019aa], introduced the concept of the boundary score statistic, adapted here for differential boundary detection. Briefly, the boundary score is calculated for each bin by sliding a window across the diagonal of the contact matrix, calculating the eigenvectors of the Laplacian matrix, finding the distance between consecutive eigenvectors (eigenvector gap) and converting them into Z-scores (boundary score, see Methods). The boundary score is a continuous measure of TAD boundary likelihood. 

In contrast to other metrics for boundary identification that rely on finding inflection points of monotonic functions, such as directionality index [@Dixon:2012aa], insulation score [@Crane:2015aa], RobusTAD score [@Dali:2017aa] (Supplementary Material), our boundary score weakly oscillates within TADs and spikes at the TAD boundary (Figure 1). This unique behavior enables easy distinction between TAD boundaries and non-TAD boundaries. An additional advantage of the boundary score is that its magnitude is directly interpretable as a "boundary strength". This is in contrast to other methods which are only interpretable relative to neighboring points. We can use this interpretability for parametric modeling of TAD boundary behavior. Our previous work has shown that the boundary score is robust to noise, sparsity, and changes in sequencing depth of Hi-C data [@Cresswell:2019aa]. Thus, the boundary score is ideal when finding differences in TAD boundaries. 

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_1}
\end{figure}
**Figure 1. Boundary score distinguishes TAD boundaries better than monotonic metrics.** TAD boundary scores calculated with four methods: directionality index, insulation score, RobusTAD and TADCompare boundary scores are shown. X-axis - Distance from TAD boundary, measured in bins (40kb each), Y-axis - Score (signed log10 values centered at zero). Results from five simulated contact matrices, 40kb resolution, with manually annotated TAD boundaries [@Forcato:2017aa] are shown.

### Differential boundary scores translate to five types of TAD boundary changes

Differential boundary score is a measure of the difference between TAD boundaries between two samples. This score follows a standard normal centered at 0 (see Methods, Supplementary Figure 1). Differential TADs are detected by finding regions with the differential boundary score is greater than 2 (Supplementary Figure 2), which intuitively corresponds to differences with a p-value smaller than 0.05. 

We divide TAD boundary changes into five categories (complex, split, merge, shifted, strength change; Figure 2, Supplementary Figure 3). A similar strategy was used in Ke et al. [@Ke:2017aa]. A TAD can be **split** between the datasets meaning it exists as a continuous TAD in one and is split into two or more TADs in another. In practice, this situation requires two shared TAD boundaries and a differential TAD between them. **Merging** is the opposite of splitting and arises when a TAD boundary surrounded by two non-differential TAD boundaries disappears in one of the contact matrices. Classification of TAD boundary change as merged and split depends on the reference contact matrix being compared to. Finally, a TAD can be split in a **complex** way meaning they are neither split or merged but instead taking on an entirely new structure. <!-- If we are comparing two contact matrices and flip the reference matrix, all splits become merges and vice-versa.--> Merged and split boundaries represent the structural change (splitting or merging) of the same TAD as opposed to complex boundaries which we consider to be part of a completely different TAD. The "complex", "merge", and "split" TAD boundaries are considered to be the most disruptive changes in the 3D structure of the genome.

A **shifted** TAD boundary is defined as the non-overlapping boundary that lies within five bins (or another user-defined threshold) of a boundary in the contact matrix in which it is being compared to. A **strength change** occurs when a TAD boundary is present in both contact matrices, but its differential boundary score magnitude is greater than the differential threshold of 2. The other cases are considered to be non-differential TAD boundaries. This framework allows us to systematically compare and classify TAD boundary changes. 

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_2}
\end{figure}
**Figure 2. Five types of TAD boundary changes.** Complex, split, and merge TAD boundary changes are considered as the major differences, while shifted and strength changes are considered as the minor differences. 

<!--### Selecting cutoffs for TAD boundary and differential boundary detection
[??? We already referred to supplementary Figure S2. This paragraph is a mess, we don't have panels C and D on that figure]
Once we have an established boundary score, TAD detection, and differential TAD detection, become a matter of choosing a cutoff that maximizes the number of true TAD boundaries detected while controlling for false detection of TAD boundaries. To this end, we tested a range of cutoffs on data with simulated noise and sparsity. The viability of each cutoff was calculated using the Youden index which is simply specificity+sensitivity-1. We find that for differential TADs, a cutoff of either 1.5 or 2 maximizes our ability to balance true and false positives (Supplementary Figure 2A & 2B). Accordingly, we choose 2 as our cutoff in this manuscript and by default in the `TADCompare` package. While a cutoff of 1.5 is viable, 2 allows us to be slightly more conservative in choosing differential TADs. In the case of normal TAD boundary detection, a cutoff of 2.5 results in the best performance at realistic levels of noise and sparsity (Supplementary Figure 2C & 2D).
-->

### TAD boundaries are highly consistent in both technical and biological replicates 

Previous studies have shown that the overlap between TAD boundaries in replicate data ranges from around 60\% to 70\% [@Dixon:2012aa; @Rao:2014aa; @Sauerwald:2018aa2]. Additionally, technical replicates have been shown to have a slightly higher proportion of shared TAD boundaries (~65%) than biological replicates (~60%) [@Sauerwald:2018aa2]. We have tested and confirmed these observations by showing that significantly more TADs were non-differential in technical replicates than in biological replicates (73% vs. 65.7%). Similarly, 9.3%/8.1% of boundaries showed significant strength change, while 7.8%/6.1% were shifted in the biological/technical replicates, respectively. A similar trend was observed for complex and merge-split boundaries. In summary, only 17.2%/12.8% of TAD boundaries were differential in biological/technical replicates, respectively (Figure 3A), confirming the higher stability of TAD structures in technical replicates.
<!--These results indicate that biological replicates are generally less stable than technical replicates as expected. Using our method, we find that the percentage of TADs which are truly differential is much lower than previous methods estimated. In total, 83% of TADs are preserved (Shifted, Strength Change, Non-Differential) in biological replicates and 87% in technical replicates. Our estimate differs from previous works which showed much less consistency among TAD boundaries. Additionally, we find that using our method, the percentage of shared TADs is consistent among different chromosomes. Among technical replicates, the percentage of preserved TADs range from 79.6% to 95.5% with a mean of 86.6% (SD = 0.07). For biological replicates the numbers ranged from 72.8% to 87.2% with a mean of 82.3% (SD = 0.07). These discrepancies suggest other approaches such as TAD calling and then using Jaccard indices to compare TADs (Dixon, Rao) or calling TADs and using an information-based metric (Sauerwald) provide an over-estimate of differentiality. Likely reasoning for these issues is a combination of the error caused by variation in TAD callers and a failure to account for small shifts in TAD boundaries which are likely due to noise. -->

### TAD boundaries are more similar within cells than tissues

Previous research showed that TADs are largely invariant across cell lines and to a lesser extent tissue types [@Pope:2014aa; @Rao:2014aa; @Schmitt:2016aa]. However, the types of TAD boundary changes remained undefined. We compared Hi-C matrices of 7 different cell-lines and 18 different tissue types [@Schmitt:2016aa] (Supplementary Table 3). In total, the average percentage of differential TAD boundaries was significantly less in cell lines (22.5%) than tissue samples (39.7%, Figure 3B). As expected, these percentages were higher than those for biological (17.2%) and technical replicates (12.8%). These results suggest that the variability of TAD boundaries mirrors the homogeneity of data types (technical replicates, biological replicates, cell lines, and tissues, in that order). 
<!--Consequently, clustering cell type- and tissue-specific Hi-C datasets using Pearson correlation of their genome-wide boundary score showed replicates from cell types clustering with each other, in contrast to tissue-specific replicates (Supplementary Figure). [??? More the current Figure 6 to supplemental] The mean Pearson correlation of boundary scores for cells was 0.89 and 0.52 for tissues. The larger variability in tissue-specific Hi-C datasets likely comes from the fact that tissue samples contain mixtures of cells while cell lines provide homogeneous samples. -->

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_3}
\end{figure}
**Figure 3. Biological replicates and cell lines have more differential TADs than technical replicates and tissues, respectively.** Differential TADs were calculated between Hi-C datasets of biological and technical replicates (Panel A, HCT-116 cell line, 50kb resolution, chr 1-22 [@Rao:2017aa]) and between cell lines and tissues (Panel B, various cell lines, 40kb resolution, chr1-22, [@Schmitt:2016aa]). Types of TAD boundary changes were recorded, and the proportions of TAD differences for each type were summarized across chromosomes. 
[??? Check figure!!! We first present replicates, then cell lines. Currently, the panels are swapped]

<!--
Overall, we observed a higher level of non-differential TAD boundaries [??? Changed from "conservation" - is this change correct?] in cell lines (Mean = 70.5%, SD = 0.08%) than in tissues (Mean = 54.6%, SD = 0.04%).

To better understand the similarity of the TAD structures across cell and tissue types, we clustered cells and tissues by Pearson correlation of their genome-wide boundary score profiles (Figure 6). The mean correlation for cells was 0.89 and 0.52 for tissues. While replicates of cell line samples were clustered with each other, this was not the case with tissues. To examine the mechanisms behind poor clustering, we found differential boundaries and calculated the percentage of each type. Replicates which failed to cluster tended to have a high level of shifted TAD boundaries. For instance, 21.2% TAD boundaries were shifted between replicates of lung samples, while 22.0% were shifted between replicates of pancreas samples. This was higher than the overall mean percentage of shifted boundaries (14.7%). All comparisons are summarized in Supplementary Table 2. These results suggest that a large percentage of inconsistency between certain tissue-based replicates may be due to small shifts in boundaries. The implications of these findings are two-fold. One is that methods which cannot identify shifted TAD boundaries will overstate the degree of difference between tissues. This is because they will incorrectly consider shifts as differential boundaries without considering the distance of boundary movement. The second is that there is a need for methods which can take a set of replicates and correct for the high number of shifted boundaries among replicates. One approach for achieving this is our proposed consensus TAD boundary calling method. With enough replicates it is expected that the consensus TAD boundary for a given region will converge to the true boundary, making shifts have less of an effect.
-->


### Each type of differential TAD boundaries is associated with different levels of epigenomic enrichment

To understand the biological relevance of types of TAD boundary changes, we identified differences between the GM12878 and IMR90 cell lines (chr1-22, 40kb resolution, [@Schmitt:2016aa]) and categorized them according to the type of change. For each change type, we assessed the number of overlapping peaks and calculated the enrichment of four genome annotation marks known to co-locate with TAD boundaries - CTCF, RAD21, insulators, and heterochromatin states.

We found that non-differential boundaries had a higher average number of overlapping peaks for all four marks, followed by "strength change" TAD boundaries (Figure 4A). Similarly, enrichment of non-differential TAD boundaries was the most significant (Figure 4B). Notably, the number of peaks for each mark was highly variable on “strength change” TAD boundaries (Figure 4A), suggesting their biological relevance is less certain. Similarly, "shifted" TAD boundaries had the lowest average number of peaks, suggesting that they may be detected due to noise and, consequently, be less biologically significant. In contrast, "complex" and "merge-split" TAD boundaries had a moderate number of overlapping peaks and were moderately enriched in them (Figure 4). These results highlight the varied biological relevance of different types of TAD boundary changes and suggest "complex" and "merge-split" changes are biologically important alterations of TAD structure. 

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_4}
\end{figure}
**Figure 4. Non-differential TAD boundaries are more enriched for selected genome annotation marks than other types of differential TAD boundaries.** Differential TAD boundariess were called between GM12878 and IMR90 cell lines and categorized based on differential boundary type. (A) Number of peaks at TAD boundaries and (B) permutation p-values (-log10) are shown. Data from [@Schmitt:2016aa], 40kb resolution, chr 1-22. 

### Each type of differential TAD boundaries is associated with distinct biological functionality

To test the biological significance of different types of TAD boundary changes, we compared mesenchymal stem cells (MSC) against neural progenitor cells (NPC) [@Schmitt:2016aa]. Altogether, we found that the vast majority of boundaries are either complex (38.6\%) or non-differential (32.6\%). Shifted (17.5\%), merge-split (7.7\%) and strength change (3.5\%) were less common (Figure 5A).  We investigated the enrichment of genes in the proximity of each type of differential TAD boundary in biological processes and other gene ontology- and pathway types using GREAT [@McLean:2010aa] (see Methods). As NPCs are more advanced on differentiation path than MSCs, we expected that TAD boundaries changed between them would be associated with genes responsible for neural development-related processes. Indeed, genes around "merge" and "complex" TAD boundary changes, as well as the "non-differential" TAD boundaries were enriched in a variety of developmental processes (e.g., "cellular developmental process", etc.), including neural-specific ("nervous system development", Figure 5A). Notably, "split" TAD boundary changes were not enriched in these processes, indicating the importance of directionality of TAD boundary changes. Genes around "merge" and "non-differential", but not "complex", TAD boundaries were enriched in differentiation-related processes (e.g., "positive regulation of cell differentiation"), while "forebrain radial glial cell differentiation" and "neural tube development" processes were exclusively enriched in genes around "merged" TAD boundaries (Figure 5A). In this case, "merge" indicates boundaries enriched in the NPC cell-line causing a separation of TADs in MSC and "split" indicates a split in NPC caused by a boundary enriched in MSC. As expected, genes around "noisy" TAD boundary changes ("shifted" and "strength change") lacked enrichment in any biological processes (Figure 5A, Supplementary Table 4). These results emphasize the importance of classifying TAD boundary changes into distinct patterns which tend to be associated with distinct biological functionality.

To further test whether different types of TAD boundary changes reflect biology of an experimental system, we used  post-auxin treatment time course experiment from Rao et al. 2017 study (HCT-116 cell line, 40kb resolution, 20, 40, 60, and 180 minutes following auxin withdrawal, 4 replicates at each time point) [@Rao:2017aa]. Auxin treatment eliminates CTCF binding genome-wide; consequently, the majority of TAD boundaries should be absent and gradually re-appear following auxin withdrawal. To identify biological processes associated with re-appearing of TAD boundaries, we compared first and last time points (20 and 180 minutes) following auxin withdrawal. As TAD boundaries were reported to be enriched in housekeeping genes [@Jin:2013aa], we expected genes around appearing TAD boundaries to be enriched in general cellular processes. Indeed, the vast majority of differential TADs were complex (41.4\%) and non-differential (34.7\%) (Supplementary Figure 4). We found that only genes around "non-differential" and "complex" TAD boundary changes showed some level of enrichment (Supplementary Figure 4, Supplementary Table 5). [??? Some tabs are empty?] As expected, "metabolic processes" and various developmental and housekeeping processes were specifically enriched in genes around complex TAD boundary changes, while cyclic AMP synthesis and metabolic processes were enriched in genes around "non-differential" TADs. From these results, we show that `TADCompare` can correctly classify non-essential TAD boundary changes ("shifted", "strength change") and detect distinct TAD boundary changes associated with shared and unique biological processes.

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_5}
\end{figure}
**Figure 5. Gene enrichment analysis of differential TADs.** Differential TADs were called between A) neural progenitor cell (NPC) and mesenchymal stem cells (MSC) (chr1-22, 50kb resolution, [@Rao:2017aa]), and B) between the first and last time point [??? Adjust, that is wrong] of auxin-treated cells from the HCT-116 cell-line (chr1-22, 40kb resolution, [@Schmitt:2016aa]. For each type of TAD boundary change, -log10-transformed enrichment p-values (`rGREAT`, see Methods) are shown as heatmaps. The top 30 gene ontology biological processes, in terms of average enrichment, are shown.

### Time course analysis framework

Time course analysis of TADs refers to the analysis of TAD boundary dynamics over time. The quantitative nature of boundary score allows us to monitor its changes at TAD boundaries across any number of time points. We recommend taking a union of TAD boundaries detected at each time point and monitor boundary score changes for each TAD boundary. Monitoring TAD boundary scores across time points provides an opportunity to quantify patterns of TAD boundary changes. 

<!-- Using boundary scores, we can also go region-by-region and identify time points which contain TADs. For each region, we use the pattern of change over time to classify them into four temporal TAD boundary categories (early appearing, late appearing, highly common and dynamic). These categories give us a simple summarization of the evolution of each TAD boundary over time. An additional function of `TADCompare` is the ability to plot TAD boundaries over time and visualize the overall trend. This gives users the ability to visually assess their data and determine the effect of time. We also give users the ability to view heatmaps giving a region-by-region visualization of changes with the type of temporal boundary type labeled. All temporal TAD functions in `TADCompare` are fully integrated with consensus boundary calling meaning if we have multiple replicates at a given time point it is possible to use consensus boundary scores instead of traditional boundary scores.  -->

Using the boundary score cutoff of 3 for TAD boundary definition, we define six patterns of temporal TAD boundary changes (adapted from [@Zhou:2019aa], Table 2, Figure 6). _Highly common_ TAD boundaries refer to boundaries present across all time points or in three out of four time points. _Early appearing_ TAD boundaries switch from non-boundary to boundary at second time points and stay as boundaries for the rest of the time points. Conversely, _early disappearing_ TAD boundaries switch from boundary to non-boundary at the second time point and stay as non-boundaries. _Late appearing_ TAD boundaries switch from non-boundaries to boundaries at the last or the second to last time point. Conversely, _late disappearing_ TADs switch from boundaries to non-boundaries at the last of the second to last time point. Finally, _dynamic_ TAD boundaries are those which have inconsistent boundary status and do not follow any of the aforementioned patterns (Figure 7). These six patterns of temporal changes can be easily adapted for a larger number of time points.

**Table 1. Six patterns of temporal TAD boundary changes.** Each column corresponds to a point in time. "1" refers to the presence of a TAD and "0" refers to the absence of a TAD at considered time point. Total column shows percentage of occurrences in CTCF degradation-recovery time course, HCT-116 cell line, chr1-22 [@Rao:2017aa].

| Temporal TAD Type  | Time Point 1 | Time Point 2 | Time Point 3 | Time Point 4 | Total (% Occurrence) |
|--------------------|:------------:|:------------:|:------------:|:------------:|:--------------------:|
| Highly Common      | 1            | 1            | 1            | 1            | 326 (17.35%)         |
|                    | 1            | 0            | 1            | 1            |                      |
| Early Appearing    | 0            | 1            | 1            | 1            | 184 (9.79%)          |
| Early Disappearing | 1            | 0            | 0            | 0            | 133 (7.08%)          |      
| Late Appearing     | 0            | 0            | 1            | 1            | 1047 (55.72%)        |
|                    | 0            | 0            | 0            | 1            |                      |
| Late Disappearing  | 1            | 1            | 0            | 0            | 79 (4.20%)           |
|                    | 1            | 1            | 1            | 0            |                      |
| Dynamic            | 1            | 0            | 1            | 0            | 110 (5.86%)          |
|                    | 1            | 0            | 0            | 1            |                      |

\begin{figure}[H]
\begin{center}
\includegraphics[width=6in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_6}
\end{center}
\end{figure}
**Figure 6. Six patterns of TAD boundary score changes across time course.** Average trajectories for each pattern of boundary score change are shown. The red horizontal line indicates the cutoff for TAD boundary detection. HCT-116 cell line, 40kb resolution, chr 1-22. 

<!-- An alternative approach to classifying boundaries is simply clustering the trend of boundary scores. First, we place each boundary score in a matrix where each column is a time point of each row is a TAD boundary. Since we are specifically interested in the trend profile, we z-normalize the boundary scores on a boundary-by-boundary basis. This normalization transforms each row into a summary of the trend over time for the corresponding boundary score. The matrix is then transformed into a Euclidean distance matrix, and hierarchical clustering with Ward's minimum variance criterion is used to assign clusters. Users can then choose their desired number of clusters and choose their own labels. -->

### Temporal TAD boundary types are associated with different levels of epigenomic enrichment

To evaluate the biological relevance of temporal patterns of TAD boundaries, we used post-auxin treatment time course experiment introduced above. Briefly, HCT-116 cells were treated with auxin to eliminate TAD boundaries, and Hi-C measures were obtained at 20, 40, 60, and 180 minutes following auxin withdrawal and subsequent TAD boundary reappearance [@Rao:2017aa]. Accordingly, we expected to detect some number of highly common TAD boundaries (already existing at 20 minutes) and boundaries appearing at different stages of post-auxin withdrawal (early/late appearing). Conversely, dynamic and early/late disappearing TAD boundaries should be rare and may potentially constitute noise in TAD boundary detection.

Boundary scores were calculated for auxin-treated cells 20, 40, 60, and 180 minutes after withdrawal. Taking the union of TAD boundaries (boundaries detected at one or more time points), we calculated temporal patterns for each boundary. We found that the vast majority of boundaries were late appearing TADs (55.7%) (Table 2, Figure 5B). Early appearing TADs (9.8%) and highly common TADs (17.3%) made up most of the other TADs present at the end of the time course. ~20% of TAD boundaries were highly common, i.e., resistant to auxin treatment, a number similar to previous works [@Nora:2017aa]. Meanwhile, 5.9% of TAD boundaries were dynamic, 7.1% were early disappearing, and 4.2% were late disappearing, highlighting potential errors in TAD boundary detection. In summary, some TAD boundaries can be detected at 20 minutes post-auxin treatment and remain present through all time points; however, the timing of TAD boundary restoration varies by TAD. 

To test whether TAD boundaries associated with different temporal patterns have different functional roles, we investigated their overlap with and enrichment in the common marks of TAD boundaries (CTCF, RAD21, insulators, heterochromatin, Figure 7A). For highly common, early- and late appearing TAD boundaries, we observed more overlaps with CTCF and RAD21 sites, insulator and heterochromatin states (Supplementary Table 7). Similarly, these types of TAD boundaries were highly enriched in the aforementioned genomic annotations (Figure 7B). Conversely, dynamic, early and late disappearing TAD boundaries showed less overlap with CTCF, RAD21, insulator and heterochromatin marks, and were less enriched in them. These observations suggest that disappearing and dynamic TAD boundaries are likely detected due to noise in the data, while TADs appearing after auxin treatment expectedly represent the biologically relevant signal.

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_7}
\end{figure}
**Figure 7. Common and appearing TAD boundaries show stronger enrichment in known epigenomic marks.** The number of peaks at TAD boundaries (A), and permutation p-values (B) within 50kb of boundaries in each temporal classification are shown. Hi-C data from [@Rao:2017aa], 50kb resolution, HCT-116 cell-line, chr 1-22.

### Temporal TAD boundary types are associated with distinct biological functionality

We further investigated these results using gene enrichment analysis (Supplementary Table 8) for temporal boundary types. We found that, with a few exceptions, all significant GO Biological pathways were enriched in late appearing TADs or early appearing TADs (Figure 5B, Supplementary Table 8), which make up the majority of TAD boundaries (Table 2, Figure 5B). Both early and late appearing TAD boundaries were enriched in metabolism-related processes, such as "cellular metabolic process", "oxidation-reduction process". Late appearing TADs, on the other hand, were enriched in "cellular component organization", "protein complex biogenesis" and the like processes (Figure 5B). These results are expected as cells may be activating metabolic and biogenesis pathways to recover after the destruction of TAD boundaries by auxin. These results confirm that `TADCompare` can accurately classify biologically relevant temporal TAD boundary changes and discern them from noisy changes. 

### Consensus boundary score for defining robust TAD boundaries across multiple Hi-C datasets 

The sizeable proportion of noisy "shifted" and "strength change" TAD boundary changes across Hi-C datasets (Figure 3) highlights the need to identify TAD boundaries that are robustly detected. The consensus boundary score, defined as the median of boundary scores across replicates, addresses this challenge. Intuitively, higher consensus boundary scores correspond to TAD boundaries supported by evidence from multiple replicates (Table 1). This is in contrast to a union of TAD boundaries, where TAD boundaries detected in at least one Hi-C dataset are pooled together. Consensus boundary scores allow us to filter out boundaries with insufficient support from multiple replicates, thus "denoise" the detected TAD boundaries. Given the fact that boundary scores are log-normally distributed (Supplementary Figure 1, Supplementary Methods), the consensus boundary scores will also be asymptotically normal. The consensus boundary score can be used as a proxy for the normal boundary score for the analysis of replicated Hi-C datasets. Consequently, the consensus boundary scores may be compared to define TAD boundary changes between groups of replicated Hi-C datasets.

**Table 2. Consensus (aka median) boundary score is supported by high boundary scores from multiple replicates.** Examples of boundary scores across five regions in three replicates, and the corresponding consensus boundary score. Both union and consensus TAD boundaries are calculated using a cutoff of 3.

|| Boundary Score 1 | Boundary Score 2 | Boundary Score 3 | Consensus Boundary Score | Union TAD Boundary?  | Consensus TAD Boundary? |
|---------|------------------|------------------|------------------|--------------------------|----------------------|-------------------------|
||         1        |         2        |         1        |              1           |             No       |                 No      |
||         3        |         2        |         1        |              2           |             Yes      |                 No      |
||         5        |         5        |         4        |              5           |             Yes      |                 Yes      |
||         3        |         3        |         3        |              3           |             Yes      |                 Yes     |
||         6        |         0        |         0        |              0           |             Yes      |                 No      |

### Consensus TAD boundaries are supported by strong biological evidence

To investigate the biological relevance of TAD boundaries defined using consensus boundary score, we defined consensus TAD boundaries across 7 cell-lines (17 matrices total) [@Schmitt:2016aa]. These boundaries represent cell type-invariant TAD boundaries supported by evidence from multiple datasets. Regions of the genome were separated into three categories based on the level of their consensus boundary score (<2, 2-4 and >4). In total, there were 65,336 non-gap [??? What's this?] bins (40kb resolution).  Expectedly, the majority (62,791 bins, 96.1% of all regions) were in the <2 category, 2,032 (3.1%) regions were in the 2-4 category, and 513 (0.8%) regions were in the >4 category. We assessed the number of overlapping peaks and the enrichment of CTCF, RAD21, insulators, and heterochromatin states in different categories of consensus TAD boundaries. Expectedly, we observed increasing average number of peaks overlapping TAD boundaries selected at more stringent consensus boundary score thresholds (Figure 8, Supplementary Table 6). Similarly, TAD boundaries with higher consensus boundary scores have stronger enrichment in genome annotations. These results suggest that TAD boundaries with higher consensus boundary scores (i.e., supported by evidence from multiple Hi-C datasets) are more biologically relevant.

<!-- Based on these results, we chose a default consensus boundary score cutoff of 3. This cutoff is used for all analyses in this work and is the default in the `TADCompare` package. In practice, the cutoff of 3 is not necessarily the only choice. For instance, a cutoff of 2 would include weaker strength TAD boundaries at the risk of more false positives and a cutoff of 4 would ensure only strong boundaries at risk of more false negatives. Due to the normality (Supplementary methods) of consensus boundary scores, we can interpret these cutoffs in terms of standard deviations from the mean boundary score. For instance, a boundary score of 3 is 3 standard deviations greater than the mean boundary score.  -->

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_8}
\end{figure}
**Figure 8. TAD boundaries defined at higher consensus boundary score thresholds show stronger overlap with and enrichment in known epigenomic marks.** TAD boundaries were classified based on the range of their consensus boundary score. Enrichment of genomic factors known to occur near TAD boundaries was calculated. (A) The number of TAD peaks within 40kb of TAD boundaries with the corresponding consensus score range and (B) the permutation p-values for each score range are shown. Data from seven cell lines, chr1-22, 40kb resolution, [@Schmitt:2016aa].

### The union of TAD boundaries is supported by weaker biological evidence than consensus boundaries

The union of TAD boundaries called in individual Hi-C datasets represents an alternative method of defining TAD boundaries across multiple datasets (Table 1). The union method may be useful for analysis of time course data, where TAD boundaries are expected to change across individual datasets. We hypothesized that the union method would select for the less biologically relevant set of TAD boundaries because many may be detected due to noise in Hi-C data. 
<!--
One advantage of the consensus boundary score is its ability to filter out weak TAD boundaries that only occur in a minority of replicates and are likely artifacts of the sequencing process. Traditionally, researchers have called TAD boundaries on individual contact matrices, taken a union of all boundaries called and then ran analyses. Contrary to what one would expect, this approach may actually reduce the quality of TAD boundary data. By adding in new replicates, we can potentially introduce noisy boundaries which are unique to each specific replicate and not biologically significant. In fact, we find a decline in quality of TAD boundaries as more replicates are added due to these noisy boundaries (Supplementary Figure 3). Consensus differs from the union method in that it automatically filters out TAD boundaries which only occur in a small portion of replicates while union calls all TADs that appear in at least one replicate (Supplementary Table 1).
-->

To evaluate the biological relevance of TAD boundaries called using both methods, we call consensus and union TADs on a set of replicates (four cell lines, 40kb resolution, 3 replicates each, data from [@Schmitt:2016aa]). Consensus scores were calculated separately for each cell line among the 3 replicates. Expectedly, the consensus method filtered out 38% of TAD boundaries (4906 vs. 3059, Supplementary Figure 5), suggesting that many TAD boundaries are detected in single datasets. We found that TAD boundaries called using consensus boundary score overlapped significantly more with CTCF sites (P = 0.0006) and RAD21 (P = 0.0002) than those called using the union method (Figure 9A). While the enrichment results were similar for consensus- and union-defined TAD boundaries, consensus TAD boundaries were more significantly enriched in "heterochromatin" (Figure 9B). Together with previous observations (Figure 6), these results strengthen our conclusion that consensus boundary scores are more effective in removing "noisy" TAD boundaries that otherwise would be captured using the union method.

\begin{figure}[H]
\center
\includegraphics[width=7in,keepaspectratio]{Finished_Figures/Final_Figures//Figure_9}
\end{figure}
**Figure 9. Consensus TAD boundaries show stronger overlap with and enrichment in known epigenomic marks than union of TAD boundaries.** (A) Number of peaks at TAD boundaries and (B) permutation p-values (-log10) are shown. Data from [@Schmitt:2016aa], four cell lines, 40kb resolution, chr 1-22.

<!--
[??? That is too much. don't report it]
An alternative, much more conservative, the approach is only to use TAD boundaries called in all or the majority of replicates or cell lines. This approach gives a much smaller set of TADs but with a reduction in noisy one-off-boundaries. To test the effectiveness of consensus boundary calling versus this method, we called TADs across 17 cell lines and 20 tissue samples and compared enrichment of genomic features at boundaries. TAD boundaries were separated based on the number of occurrences and binned into 3 categories (1-5, 6-10, and 11-20 occurrences). Among cells, we find a non-significant increase in enrichment of CTCF, RAD21, and heterochromatin and a significant increase in insulators (P = 0.0460) between consensus and the highest quality 11-20 group. For tissues we find a significant increase in CTCF, RAD21, heterochromatin and insulators (P = <0.0001, P = <0.0001, P = 0.0002, P = 0.0006) between consensus and the 11-20 group. In general, we find that TADs occurring in a large number of tissue samples or cell lines have higher enrichment than more rare TAD boundaries suggesting an even stronger need for filtering out weak boundaries. Consensus boundary score also allows us to get a much more refined set of TADs in cells (1158 vs. 1268) and tissues (119 vs. 557) when compared with taking the majority of occurrences. These results show consensus boundary score is a powerful method for summarizing TADs across many different sources and provides an improvement over naive methods such as using a union of TADs based on number of occurrences (Supplementary Figure 4).
-->

### Runtime performance of TADcompare

When ran on data from [@Rao:2014aa], without parallelization, both consensus TAD calling and differential TAD detection were exceptionally fast. In total, for the entire genome, differential TAD detection took ~6 seconds on 100kb data, ~9 seconds on 50kb data, ~17 seconds on 25kb data and ~312 seconds on 10kb data. In the case of consensus TAD calling, `TADCompare` took ~17 seconds to run on 50kb data for 4 matrices, ~32 seconds for 8 matrices and ~45 seconds for 12 matrices. On 10kb data, it took ~611 seconds to run for 4 matrices, ~1152 seconds for 8 matrices and ~1680 seconds for 12 matrices. For a full summary of runtimes across all resolutions, see Supplementary Figure 6. 

# Discussion

The initial development of Hi-C technologies focused on investigating individual genomes. While several key properties have been discovered (chromosome territories, A/B compartments, TADs, chromatin loops), the next steps include investigating changes in the 3D structure across multiple conditions. We [@Stansfield:2018aa;@Stansfield:2019aa] and others [@Djekidel:2018aa;@Lun:2015aa] started to develop methods for comparative analysis of the 3D structures. However, to our knowledge, no methods are available for differential analysis of TAD boundaries. In this work, we introduce a method for differential TAD boundary analysis, including time course, that supports replicated Hi-C data. The method is based on a novel boundary score metric that provides a continuous measure of TAD boundary likelihood [@Cresswell:2019aa]. We introduce unique terminology for classifying differential and temporal TAD boundary changes. We show that our approach is robust and effective at identifying distinct biology associated with different types of TAD boundary changes. Our method is implemented in the `TADCompare` R package available on Bioconductor, filling a vital gap in intuitive R-based software for TAD detection and comparison. 

The boundary score concept developed in our work addresses three main problems: differential TAD boundary detection, time course analysis of TAD boundary changes, and consensus TAD boundary calling. Yet, it has a wider scope of applications. Future work will expand the utility of boundary score by developing a similarity/reproducibility score to measure the agreement between (multiple) Hi-C matrices, in the same vein as HiCRep [@Yang:2017aa],<!-- which uses spectral methods and other similarity score metrics such as--> Selfish [@Ardakany:2019aa], GenomeDISCO [@Ursu:2017aa], HiC-Spector [@Yan:2017aa], QuASAR-Rep [@Sauria:2015aa]<!-- which use different approaches-->. Furthermore, for differential TAD boundary detection, our method is still limited to the comparison of two profiles of (consensus) TAD boundary scores. This approach will eventually be expanded to include comparisons of many contact matrices, similar to the concept of comparing groups of multiple replicates in RNA-seq data. Finally, there is still room for expansion of time course boundary analysis. The continuous nature of boundary score allows for adopting time course analysis methods developed for gene expression studies [@Bar-Joseph:2012aa]. More flexible classification of temporal trends may be considered, such as 24 temporal patterns proposed by Zhou et al. 2019 [@Zhou:2019aa], or fuzzy clustering techniques that do not require a pattern to belong to a specific cluster [@Abu-Jamous:2018aa]. In summary, our work enables further development of various aspects of 3D genome analysis.

One difficulty in our work is how to accurately quantify the biological relevance of TAD boundaries (differential, time-varying, and consensus) that we detect. There is no natural gold standard for TAD boundaries, but there are known genomic features that form the building blocks of TADs (CTCF, RAD21). In practice, we can use enrichment near boundaries of these as a proxy for "true boundaries". To test whether enrichment is different than random (non-boundaries), we use a permutation test and present these p-values. However, we can not compare these values between groups because the variance of random samples is affected by sample sizes. As a result, all p-values presented in the manuscript must be considered only within the context of the boundary type itself and not compared between boundary types or resolutions.

Our results in this manuscript demonstrate the ability of `TADCompare` to provide accurate, biologically relevant results. The methods implemented span differential, time-course, and consensus analysis. To date, `TADCompare` is the only actively maintained and publicly available tool to provide any of this functionality. We intend for `TADCompare` to be a one-stop tool for comparison of HiC datasets, providing simple, easy-to-interpret, results promptly. As a one-of-a-kind tool, `TADCompare` will increase the ability of researchers to extract important biological insights from the structure of TAD boundaries.


# References

<div id="refs"></div>

\newpage

# Supplementary Legends

**Supplementary Figure 1. Log-normal distribution of eigenvector gaps converted to boundary Z-scores.** Eigenvector gaps were calculated for contact matrices across three resolutions (10kb, 25kb, and 50kb, Hi-C data from [@Rao:2014aa], GM12878 cell line, chr 1-22). Density plots are shown for the (A) Natural log of the eigenvector gaps and (B) Boundary scores derived from the same data, separated by resolution. Regions of non-TADs are highlighted by a yellow bar, moderate strength TADs (2 < boundary score cutoff < 3) are highlighted by a red bar, and strong TADs (cutoff > 3) are shown using a green bar. We see a slightly right-skewed distribution due to the filtering of gaps for plotting purposes.

**Supplementary Figure 2. Window size of 15 units of Hi-C data resolution and TAD boundary score cutoff of 2 yields consistent TAD boundary detection.** Differential TAD boundaries were compared between two simulated data sets with window size sizes ranging from 10 to 25 and boundary score cutoff ranging from 1.5 to 4. Youden index (balanced sensitivity and specificity metric) was calculated for each combination and plotted to show agreement with ground-truth annotations. Results are shown for noise-injected matrices (A) and sparsity-injected matrices (B).

**Supplementary Figure 3. Visualization of different types of boundary score patterns.** Patterns of raw boundary scores are shown for 5 different types of differential boundaries (Merge, split, complex, shifted, and strength change). The red horizontal line corresponds to the minimum cutoff for a TAD boundary (40kb resolution, human neural progenitor cell line, data from [@Schmitt:2016aa]). Data from chromosome 22 with the most representative examples chosen.

**Supplementary Figure 4. Heatmap of gene ontology enrichment at the first and last time point in auxin-treated data.** Differential boundary identification was performed on auxin-treated data at the time of application (first time point) and complete withdrawal (last time point) (HCT-116 cell line, chr1-22, 40kb resolution, [@Schmitt:2016aa]). A barplot of the proportion of each boundary type (A) and FDR-adjusted hypergeometric p-values (B) obtained from gene ontology enrichment analysis using `rGREAT` (See methods) are shown. The top 30 pathways, in terms of average enrichment, are shown and clustered using Ward clustering.

**Supplementary Figure 5. Venn diagram of union and consensus TAD counts.** Consensus and union TADs were called across four different cell lines (hesc, mesynchymal, npc, trophectoderm) and the number of union and consensus TADs were recorded. The Venn diagram shows the complete overlap of consensus TADs within union TADs. (40kb resolution, data from [@Schmitt:2016aa])


**Supplementary Figure 6. Runtime of TADCompare.** Plot containing the runtime of two-way comparison (A) and consensus TADs called on 4, 8, 12, and 16 replicates (B). Each point represents the runtime for a specific chromosome. X-axis - Chromosome, Y-axis - Runtime in seconds. Hi-C data from [@Rao:2014aa], chr 1-22, 10kb, 25kb, 50kb, and 100kb resolution.

**Supplementary Table 1. Contact matrix data sources.** The source of all contact matrices, experimental and simulated, used in this paper are provided. Experimental data are separated based on the study and cell line.

**Supplementary Table 2. Genomic annotation data sources.** The sources, with download links, for all genomic annotation used in this paper are included.

**Supplementary Table 3. Summary of differential boundary types across tissues and cell-lines.** The percentage of each type of differential boundary for all tissue-tissue and cell line-cell line comparisons is reported. Results are aggregated over all chromosomes. Hi-C data from Schmitt et al. [@Schmitt:2016aa], 40kb resolution, chr 1-22.

**Supplementary Table 4. Gene ontology enrichment for differential boundary types.** Differential boundaries were identified between the neural progenitor cells (NPC) and mesenchymal stem cells (MSC) [@Schmitt:2016aa]. Pathway analysis was performed using `rGREAT` (Methods), and results are separated by ontology. Boundaries with an FDR adjusted p-value of <0.3 are shown. 40kb resolution, chr1-22.

**Supplementary Table 5. Gene ontology enrichment between the first and last time point in auxin-treated data .** Differential boundaries were identified between the first and last time point of auxin-treated data [@Rao:2017aa]. Pathway analysis was performed using `rGREAT` (Methods), and results are separated by ontology. Boundaries with an FDR adjusted p-value of <0.3 are shown. 50kb resolution, chr1-22.

**Supplementary Table 6. Enrichment across different temporal boundary types.** Temporal boundary types were identified across four time points in auxin-treated data [@Rao:2017aa]. Results are shown for four types of temporal TAD (Early Appearing, Late Appearing, Highly Common, Dynamic). Permutation p-values, along with enrichment or depletion designations, are reported. HCT-116 cell line, 40kb resolution, chr 1-22.

**Supplementary Table 7. Gene ontology enrichment for different temporal boundary types.** Temporal boundary types were identified across four time points in auxin-treated data [@Rao:2017aa]. For each temporal boundary type, pathway analysis was performed using `rGREAT` (Methods), and results are separated by ontology. Boundaries with an FDR adjusted p-value of <0.3 are shown. HCT-116 cell line, 50kb resolution, chr1-22.

**Supplementary Table 8. Enrichment across different consensus scores.** Consensus scores were called across 17 contact matrices representing 7 different cell lines. Results were dichotomized into three groups (<2, 2-4, >4) based on consensus boundary scores. Permutation p-values, along with enrichment or depletion designations, are reported. Hi-C data from Schmitt et al. [@Schmitt:2016aa], 40kb resolution, chr 1-22. 

# Acknowledgements

This work is supported in part by the PhRMA Foundation Research Informatics Award. The authors would like to acknowledge Dr. Katarzyna Tyc for helpful feedback.

_Conflict of Interest._ None.

# Author Contributions Statement

MGD and KGC conceived the project, KGC implemented `TADcompare` and wrote the analysis scripts. MGD and KGC wrote the manuscript.

# References
