---
bibliography: [/Users/mdozmorov/Documents/Work/VCU_grants/1_Dozmorov/networks.bib,/Users/mdozmorov/Documents/Work/VCU_grants/1_Dozmorov/brain.bib,/Users/mdozmorov/Documents/Work/VCU_grants/1_Dozmorov/enrichment.bib,/Users/mdozmorov/Documents/Work/VCU_grants/1_Dozmorov/3D_refs.bib]
csl: /Users/mdozmorov/Documents/Work/presentations/Templates/grant_template/styles.ref/genomebiology.csl
output:
  word_document:
    reference_docx: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.doc/Arial_11_single_space_normal_margins.docx
  pdf_document: default
  html_document: default
editor_options: 
  chunk_output_type: console
---

# Coherence of Molecular Mechanisms in Genes Identified Through GWAS In Major Human Disease and Traits

Dozmorov, Mikhail G.^1+^ (mikhail.dozmorov@vcuhealth.org)  
Cresswell, Kellen G.^1+^ (cresswellkg@vcu.edu)  
Reimers, Mark^2^ (reimersm@msu.edu)  
Bacanu, Silviu-Alin^3^ (silviu-alin.bacanu@vcuhealth.org)  
Craver, Carl^4^ (ccraver@wustl.edu)  
Kendler, Kenneth S.^3^ (kenneth.kendler@vcuhealth.org)  

^1^ - Department of Biostatistics, Virginia Commonwealth University   
^2^ - Neuroscience Program and Dept. Biomedical Engineering Michigan State  
^3^ - Virginia Institute for Psychiatric and Behavior Genetics and the Department of Psychiatry, Virginia Commonwealth University  
^4^ - Philosophy-Neuroscience-Psychology Program, Washington University in St. Louis  
^+^ - equal contribution

# Abstract

**Background.** Complex phenotypes such as height and intelligence, are thought to be a product of the collective effects of multiple phenotype-associated genes, or, more precisely, interactions among their protein products. High/low degree of interactions is suggestive of coherent/random molecular mechanisms, respectively. Comparing the degree of interactions may help to better understand the coherence of the molecular mechanisms underlying complex phenotypes and the potential for therapeutic intervention. However, direct comparison of the degree of interactions is difficult due to differences in size and configuration of phenotype-associated gene networks.

**Methods.** We introduce a measure of network coherence as a ratio of internal vs. external distributions of the degree of interactions. The internal degree distribution is defined by interaction counts within a phenotype-specific gene network, while the external degree distribution counts interactions with other genes in the whole protein-protein interaction (PPI) network. We present a novel method for normalizing the internal degree distributions, making them directly comparable. 

**Results.** Using STRING and BioGrid PPI databases, we compared the coherence of 133 phenotype-associated gene sets from GWAScatalog against size-matched KEGG pathways (the reference for high coherence) and random networks (the lower limit of coherence). We observed a range of coherence estimates for each category of phenotypes. Metabolic traits and diseases were the most coherent, while psychiatric diseases and intelligence-related traits were the least coherent. 

**Conclusions.** We present a general-purpose method for estimating and comparing the coherence of phenotype-associated gene networks that accounts for the network size and shape differences. Our results highlight the gaps in of our current knowledge of the genetics and molecular mechanisms of complext traits and suggest priorities for future work.

**Keywords:** GWAS, network, degree, coherence

# Introduction

Genome-wide association studies (GWAS) have significantly advanced our understanding of complex phenotypes by identifying disease- and trait-associated genetic markers and suggesting corresponding genes [@Altshuler:2008aa; @Risch:1996aa]. However, GWAS findings explain only a fraction of heritability [@Eichler:2010aa; @Maher:2008aa; @Manolio:2009aa]. Furthermore, the corresponding genes are often spread across different chromosomes with no known connection to one another, hindering understanding of the molecular mechanisms. These limitations of current generation GWAS might stem from the yet incomplete knowledge of the genetic determinants of complex phenotypes, the potential heterogeneity of complex phenotypes and/or the causes other than genetics [@Botstein:2003aa; @turkheimer2012genome; @Sullivan:2018aa].

Many studies have shown that genetically-driven complex phenotypes are often associated with functionally related genes that are more likely to form networks of interacting protein products [@Hartwell:1999aa; @Krauthammer:2004aa; @Huang:2006aa; @Lage:2007aa; @Cordell:2009aa; @Emily:2009aa; @Feldman:2008aa]. This observation has been verified systematically for a large number of diseases [@Menche:2015aa], thus confirming a fundamental hypothesis of the interactome-based approach to human phenotype, namely that disease- and trait-associated genes tend to form interaction modules [@Goh:2007aa]. Such interaction modules may be viewed as connected subnetworks within the full interactome and may contain all molecular determinants of a certain phenotype. Consequently, methods to identify phenotype-associated networks of functionally-related genes have been developed [@Krauthammer:2004aa; @Cordell:2009aa; @Akula:2011aa]. 

Network properties, such as connectivity (aka degree distribution) [@Ghiassian:2015aa; @Lee:2008aa; @Barabasi:2004aa; @Feldman:2008aa], can be compared to better understand the relationships among the phenotypes [@Vidal:2011aa; @Barabasi:2011aa; @Wang:2011aa]. Networks formed by phenotype-associated genes, or, more precisely, their protein-protein products, are thought to have high connectivity [@Huang:2006aa; @Krauthammer:2004aa; @Goh:2007aa]. This intuition reflects a well-known "guilt-by-association" principle that genes (or, more precisely, their products) that form an interaction network are more likely to share similar functions [@Farber:2013aa], or co-expression patterns [@Huang:2006aa; @Michalak:2008aa]. Although this view has been criticized [@Gillis:2012aa], several studies consistently observed phenotype-associated genes to be either connected as a single network or to participate in common phenotype-specific subnetworks [@Menche:2015aa; @Lee:2011aa; @Akula:2011aa; @Ideker:2008aa; @Iossifov:2008aa]. Therefore, high connectivity among phenotype-specific genes may indicate coherent molecular mechanisms that could be targeted therapeutically. In terms of graph theory, we are asking whether a phenotype-specific network is a community - a highly connected subgraph relatively well-separated from the rest of the network [@fortunato2010community]. However, the direct comparison of connectivity across phenotype-specific networks is hindered by the fact that different phenotypes are associated with different numbers of genes forming networks of different sizes and configurations.

<!--This study compares the connectivity of networks formed by disease- or trait-associated genes and gene products. We hypothesized that genes associated with (categories of) diseases share common molecular mechanisms and, hence, form connected (i.e., coherent) networks. Therefore, the level of connectivity may inform us about similarities and differences among the diseases. The terms “network connectivity,” “coherence,” or “average degree distribution” are used interchangeably as appropriate. We developed a method for normalizing the internal network connectivity that accounts for differences in network sizes formed by disease-associated genes. We compared the ratio of normalized internal vs. external network connectivity between psychiatric, neurologic, autoimmune, cancer, cardiovascular, metabolic and other diseases and traits. We used KEGG pathways [30] as a reference for a coherent network with common molecular mechanisms, and we used sets of randomly sampled genes as a reference for the absense of coherence. We found that genes associated with metabolic diseases and traits and autoimmune diseases formed coherent networks similar to KEGG pathways. On the contrary, the coherence of networks of psychiatric and neurologic disease-associated genes, as well as intelligence-associated traits, was similar to that of randomly selected genes. Cancer, anthropometric traits, eye-, and cardiovascular diseases and traits formed networks of intermediate connectivity. Our study identifies significant differences between categories of diseases in terms of network properties of the disease-associated genes and highlights the limitations of the current understanding of the molecular determinants of psychiatric disorders.-->
This study presents a novel measure of network coherence as a ratio of the internal vs. external degree distributions. The internal degree distribution is defined as gene-specific interaction counts within a protein-protein interaction (PPI) network formed by phenotype-associated genes, while the external degree distribution counts interactions with other genes in the whole PPI network. We used KEGG pathways [@Kanehisa:2000aa] as a reference for a paradigmatically high-coherent network [@Lee:2003aa], while sets of randomly sampled genes were used as the reference for the absence of coherence. Using KEGG and random networks matched in size to a phenotype-specific network, we derived normalized measure of coherence that can be compared across phenotypes. We hypothesized that the level of coherence may inform us about similarities and differences among phenotypes. Using two PPI databases (STRING [@Szklarczyk:2015aa] and BioGrid [@Chatr-Aryamontri:2017aa]), we compared coherence of 133 phenotypes from GWAScatalog [@Hindorff:2009aa; @MacArthur:2017aa]. Our results show the tendency of complex phenotypes, such as intelligence-related traits, to have low coherence and metabolic traits and diseases to have high coherence. Our method enables direct comparison of coherence measures and highlight gaps in current understanding of the molecular mechanisms of many phenotypes, e.g., schizophrenia being the lowest coherence disorder in the already low-coherence "Psychiatric disease" category.

# Results

## Phenotype-specific networks of protein-protein interactions (PPIs)

We collected 133 phenotype-associated gene lists from the NHGRI-EBI GWAS catalog [@MacArthur:2017aa] (see Methods). These were grouped into seven disease categories (“Autoimmune”, “Cancer”, “Cardiovascular disease”, “Eye disease”, “Metabolic disease”, “Neurologic”, “Psychiatric”) and five trait categories (“Anthropometric trait”, “Cardiovascular trait”, “Eye trait”, “Intelligence”, “Metabolic trait”, Table 1, Supplementary Table 1). 

We created networks of phenotype-associated genes using PPI information from STRING [@Szklarczyk:2015aa] and BioGrid [@Chatr-Aryamontri:2017aa] databases (see Methods). We opted for the use of three types of PPI data. First, STRING data filtered by the interaction confidence score above 500 (middle of the bimodal distribution of the score, 0-1000 range, referred hereafter as “STRING filtered”) was used as the primary source of curated PPI data. This filtering step ensured that only high-confidence PPIs were selected. The advantage of using filtered data is the reliance on high-quality PPIs. The disadvantage is that the number of PPIs may be insufficient for forming phenotype-associated networks; consequently, phenotypes with genes without PPIs are removed from the analysis (see Methods). Second, complete STRING data were used to maximize the use of PPI information at the expense of potential noise (referred hereafter as “STRING”). The advantage of using complete data is that more trait-associated genes will have PPI information and, hence, can be analyzed. The disadvantage, however, is that the results may be less reliable due to the presence of noisy PPIs having low confidence scores. Third, BioGrid PPI data, which has been curated to keep only high-confidence interactions, were used. The use of different data sources (STRING and BioGrid) and filtering (complete or filtered STRING data) was intended to increase the generalizability of our conclusions.

<!--Tables/table_1.csv-->
**Table 1. Summary statistics of the analyzed categories.** Average coherence estimates are shown. "NA" values indicate that a category lacked phenotypes with sufficient number of PPIs. Sorted by "String Filtered" average coherence.

| Category               | Total Phenotypes | Number of SNPs | Number of Genes | Biogrid | String | String Filtered |
|------------------------|:----------------:|:--------------:|:---------------:|:-------:|:------:|:---------------:|
| Disease                |                  |                |                 |         |        |                 |
| Metabolic disease      |         5        |       345      |       280       |  0.515  |  0.776 |      0.845      |
| Autoimmune             |        20        |      1415      |       1212      |  0.470  |  0.801 |      0.702      |
| Eye disease            |         8        |       270      |       242       |  0.300  |  0.509 |      0.654      |
| Neurologic             |         8        |       382      |       343       |  0.710  |  0.498 |      0.637      |
| Cancer                 |        18        |       713      |       610       |  0.548  |  0.658 |      0.604      |
| Cardiovascular disease |         9        |       336      |       289       |  0.130  |  0.577 |      0.602      |
| Psychiatric            |        11        |       678      |       611       |  0.560  |  0.528 |      0.583      |
| Trait                  |                  |                |                 |         |        |                 |
| Metabolic trait        |        10        |       790      |       646       |  0.766  |  0.779 |      0.842      |
| Eye trait              |         5        |       158      |       109       |    NA   |  0.678 |      0.698      |
| Anthropometric trait   |        19        |      2143      |       1822      |  0.600  |  0.624 |      0.678      |
| Cardiovascular trait   |        15        |       728      |       604       |  0.463  |  0.655 |      0.537      |
| Intelligence           |         5        |       236      |       198       |  0.310  |  0.334 |      0.373      |

## Degree distribution as a measure of coherence

A typical network consists of nodes connected by edges [@Barabasi:2004aa]. In genomics, nodes typically represent genes and edges correspond to some measure of interactions, e.g., gene co-expression or interactions between protein products [@Goncalves:2012aa; @Wang:2012aa]. In terms of phenotypes, an intuitive expectation is that they will be represented by coherent networks of functionally related genes, similar gene expression profiles, shared genomic variants, higher PPI interactions, higher co-morbidity [@Zhang:2010ab; @Goh:2007aa; @Driel:2006aa; @Wang:2012aa; @Hamaneh:2015aa; @Menche:2015aa; @Jia:2011aa], reviewed in [@Dozmorov:2018aa; @Wang:2011aa]. In terms of graph theory, a coherent network is a community consisting of a group of nodes that are relatively highly connected to each other but sparsely connected to other nodes in the network [@fortunato2010community; @Radicchi:2004aa]. 

Numerous network properties have been defined to describe network structures [@Barabasi:2011aa; @Wang:2012aa]. A degree of a node, or connectivity, is one of the most fundamental characteristics defining the number of other nodes connected to a given node. A collection of degrees of all network nodes form a degree distribution. Comparing degree distributions among networks is an intuitive way to gain an understanding of network similarities and differences in terms of connectivity (aka coherence) of the corresponding genes [@Barabasi:2004aa; @Feldman:2008aa]. We hypothesized that comparing degree distributions of networks formed by phenotype-associated gene sets may inform us about similarities and differences among phenotypes in terms of coherence estimates, informing us about the underlying molecular mechanisms.

## Internal vs. external degree distributions as a measure of the coherence

<!-- inspired by previous [@Menche:2015aa] and our (Supplementary Figure S3) observation that network size and configuration are irrelevant when considering interactions within the whole PPI network. -->
Given that phenotype-associated networks occur within the global network of PPIs, we developed a ratio-based metric of coherence. This metric is inspired by previous work showing that communities within a large network have high internal but low external levels of connectivity [@PhysRevE.81.046110; @Spirin:2003aa]. Thus, for each phenotype-associated PPI network, we considered the relationship between its internal and external degree distributions. The internal degree distribution is defined by considering edges within an isolated network formed by phenotype-associated genes. The external degree distribution is defined by considering edges to other genes in the whole PPI network. Intuitively, a highly coherent network is expected to have a large number of internal edges but a relatively small number of external edges. An extreme example of high coherence would be a fully internally connected network with zero interactions with external genes. Conversely, a low-coherence network is expected to have a relatively low number of internal edges compared to external edges, mirroring a random network. 

To compare coherences, we measured coherence as the slope of a line through a scatterplot of internal vs. external degree distributions. That is, we visualize internal (X-axis) vs. external (Y-axis) degrees for each phenotype-specific gene on a single plot and fit a regression line through it. An example of a network with the highest coherence (full internal connectivity, zero external connections) would be represented by a horizontal line. Conversely, an example of a network with the lowest coherence would be a set of non-interacting genes (zero internal connections and full external connectivity) would be represented by a vertical line. Consequently, slopes of internal vs. external degree distributions of phenotype-associated networks would represent the phenotype-specific levels of coherence. In summary, comparing internal vs. external degree distributions represents a viable metric to quantify and compare the molecular coherence of phenotype-associated networks.

## Networks of KEGG pathways and randomly selected genes as reference for high and low coherence, respectively

The coherence of phenotype-associated gene sets should be measured with respect to the reference level of networks with paradigmatically high and low coherence. To establish a reference of high coherence, we considered networks from collections of the Molecular Signatures Database (MSigDB v6.2) [@Subramanian:2005aa]. MSigDB contains sets of genes having various types of functional relationships. We found that pathways from the Kyoto Encyclopedia of Genes and Genomes (KEGG) [@Kanehisa:2000aa] had the smallest average slope corresponding to the highest level of coherence (Supplementary Table S2). Networks assembled from Gene Ontology - Cellular Component (GOCC) collection and REACTOME pathways followed, representing alternative sources of networks of high coherence. This is expected as genes expressed in the same cellular components (GOCC genes) or participating in the same metabolic pathwaays (REACTOME genes) are presumed to interact more frequently. In our study, KEGG networks were used as the primary reference of high coherence, in parallel with GOCC and REACTOME collections. Conversely, as a reference for low coherence, we randomly sampled genes from a pool of genes in a given PPI database. These random gene lists represent expected network coherence arising by chance. The slopes of KEGG and random networks represent reference levels for high and low coherence, respectively.

## Network size and configuration hinders direct comparison of degree distributions

Direct comparison of degree distributions of the networks formed by phenotype-associated genes is hindered by the fact they depend on network size and configuration [@Leskovec:2010:ECA:1772690.1772755]. An intuitive example is to consider a ring and a fully connected network of size 10 and 20 nodes, respectively (Figure 1). The degree distributions of the ring networks can be directly compared regardless of network size (vectors of "2" of length 10 and 20, respectively, Figure 1A).  Although the fully connected networks are similar in that they are fully connected (i.e., similarly coherent), their degree distributions differ (all nodes in the 10-node network have degree "9", while all nodes in the 20-node network have degree "19", Figure 1B). <!--Normalizing the degree distributions of the fully connected networks by dividing them by network size makes them identical as would be expected for networks with similar coherence. However, similar normalization of ring networks will make their degree distributions different (0.22 for the 10-node network and 0.11 for the 20-node network), leading to the illusory impression of differences in coherence between intuitively equally coherent networks. -->These observations highlight the difficulty in comparing degree distributions of phenotype-associated gene networks due to two facts: 1) network sizes differ, and 2) network configuration (fully connected, ring, or intermediate connectivity) is unknown.

+-------+
| ![](Figures/Fig1_network_distributions.png)      |
| **Figure 1. Degree distributions are affected by network size and configuration.** Degree distributions of A) ring networks and B) fully connected networks containing 10 and 20 nodes. While degree distributions of the similarly coherent ring networks can be directly compared, degree distributions of fully connected networks depend on network size.   |
+-------+

To observe whether network properties affect coherence estimation in experimental settings, we investigated the effect of network size on the slopes of KEGG and random networks. We observed negative correlation of network size with the slopes of KEGG and random networks (-0.58/-0.86, respectively, Figure 2, Supplementary Table S2). These observations confirm the notion that coherence estimation using internal vs. external degree distributions should be controlled for network size.

+-------+
| ![](Figures/Fig2_Size_Slope_Plot.png)      |
| **Figure 2. Network size is inversely associated with coherence.** Slopes of internal vs. external degree distributions (aka coherence) for KEGG and random networks are plotted against network size.   |
+-------+

## Network size-independent estimation of coherence

Given the dependence of internal and external degree distributions on network size, we designed a simple strategy to account for it. Briefly, for each phenotype-specific network of a given size, we create size-matched references of high and low coherence. Specifically, we select the ten KEGG pathways closest in size and ten random networks matched in size as references for high/low coherence for the corresponding phenotype. Consequently, we use the slopes of internal vs. external degree distributions created by these size-matched references to normalize the slope of the phenotype-specific network to the [0, 1] range (see Methods), where 0/1 correspond to random/high coherence, respectively. We found that the use of size-matched references indeed removes the dependency between coherence and network size (Supplementary Figure S1) and allows for direct comparison of normalized coherences across phenotypes.

## Results derived from different PPI data are largely consistent

The potential drawback of this reference size-matching approach is that phenotypes with large network sizes (e.g., "Obesity-related traits", 779 genes) may not have the size-matched reference of high coherence (e.g., the largest KEGG pathways is 374 genes). Consequently, coherence estimation for such networks may be affected by the smaller size of reference of high coherence. Therefore, we use networks from GOCC and REACTOME collections that have larger maximum number of genes (Supplementary Table S2) in parallel with KEGG networks as a reference of high coherence. We found that coherence estimations using KEGG, GOCC, and REACTOME networks as references of high coherence are highly concordant (Average Pearson correlation coefficient > 0.59, Supplementary Figure S2). The use of size-matched networks from KEGG, GOCC, and REACTOME collections enables robust estimation of coherence for phenotype-specific networks of different sizes. Given the similarity of the results obtained with different PPI databases and references of high coherence, we present results obtained with STRING filtered PPI database and KEGG networks as a reference of high coherence, with other observations presented in Supplementary Table S3 and Supplementary Figures S3-S8.

## Coherence of all phenotypes significantly differs from random

To estimate the significance of coherence measures, we developed a permutation-based approach (see Methods). Notably, we observed coherence for all diseases to be highly significantly different from random (permutation p-value < 0.002, Supplementary Table S3). Similarly, the distribution of coherence estimates was skewed towards 1.00 (highest coherence), with the median coherence being 0.86. Among traits, "Corneal structure", "Body mass index", "Weight" were the most coherent (1.02, 1.00, 0.99, respectively). [???]  

"Obesity-related traits" and "Height" were on the lowest range of coherence estimates (0.42 and 0.54, respectively). "Weight", "Telomere length", and "Corneal structure" were on the highest range of coherence (1.00, Supplementary Table S3). These results indicate that coherence of phenotype-specific networks is generally high and significantly different from random.

## Metabolic and intelligence-related traits as examples of high/low coherence

We estimated coherence of 54 traits from five categories (Figure 3, Supplementary Table S3, Supplementary Figures S3-S5). Among anthropometric traits, "Weight" and "Body Mass Index" had the highest level of coherence. Other examples of traits with high coherence include blood pressure-related traits (cardiovascular) and glucose- and cholesterol-related traits (metabolic). Notably, metabolic and eye trait categories had the highest overall coherence (mean coherence 0.90 and 0.93, respectively). Several notable examples of low coherence include "Obesity-related traits", "Height", and "Longevity" (anthropometric traits), blood count-related traits (cardiovascular). Traits in the "Intelligence" category had low overall coherence (mean coherence = 0.76). These results highlight the relatively good knowledge of the genetics of metabolic traits which corresponds to highly coherent networks. Conversely, current knowledge about the genetics of intelligence-related traits appear insufficient as reflected in the low coherence of the corresponding networks.

+-------+
| ![](Figures/Fig3_summary_traits.png)      |
| **Figure 3. Coherence estimates of traits.** Size of dots represents the level of normalized coherence (X-axis) for individual traits (Y-axis).   |
+-------+

## Metabolic and neurologic diseases have high coherence

We further estimated coherence of 80 diseases from seven categories (Figure 4, Supplementary Figures S6-S8, Supplementary Table S3). Metabolic diseases had high overall coherence (average coherence 0.93), with "Obesity" and "Type 2 diabetes" being the most coherent (0.99 and 0.94, respectively). Similarly, the average coherence for neurologic diseases was 0.86, with "Alzheimer's disease" and "Parkinson's disease" being highly coherent (0.94 and 0.92, respectively). The high coherence of metabolic and neurologic disease networks highlight the relatively good knowledge of their genetics.

## Intermediate coherence of autoimmune, cancer, and eye diseases

"Rheumatoid arthritis", "Ankylosing spongylitis", "Psoriasis" were the most coherent in the "Autoimmune" category (0.94, 0.91, 0.89, respectively). On the other end of the coherence spectrum, "Multiple sclerosis" and "Allergic rinitis" were the least coherent (0.76 for both). The average coherence of autoimmune diseases was high, 0.85, with all but two autoimmune diseases having high coherence (> 0.80). Diseases in the "Cancer" category were similarly highly coherent (average coherence 0.84). "Non-small lung cancer", "Testicular germ cell tumor", "Epithelial ovarian cancer" were the most coherent (0.94, 0.91, 0.88, respectively) while "Basal cell carcinoma", "Breast cancer" "Hodgkin's lymphoma" were the least coherent (0.64, 0.78, 0.78, respectively). Overall, networks of all but "Basal cell carcinoma" cancers had high coherence (> 0.78). Diseases in the "Eye disease" category were more heterogeneous, with coherence ranging from 0.71 for "Myopia (pathological)" disease to 0.97 for "Refractive error". The average coherence of diseases in the "Eye disease" category was relatively high (0.86). These results suggest good overall knowledge of the genetics of autoimmune, cancer, and eye-related diseases and highlight limited knowledge for some diseases in these categories.

+-------+
| ![](Figures/Fig4_summary_diseases.png)      |
| **Figure 4. Coherence estimates of diseases.** Size of dots represents the level of normalized coherence (X-axis) for individual diseases (Y-axis).   |
+-------+

## Psychiatric and cardiovascular diseases as examples of low coherence

Diseases in "Psychiatric" and "Cardiovascular" categories showed the lowest average level of coherence (0.82 and 0.83, respectively). The corresponding coherence estimates were highly heterogeneous. Among cardiovascular diseases, "Atrial fibrillation", "Coronary artery disease" showed the highest level of coherence (0.98 and 0.89, respectively), while "Sudden cardiac arrest" and "Hypertension" had the lowest coherence (0.70 and 0.77, respectively). Among psychiatric disorders, "Smoking behavior" and "Alcohol dependence" had high coherence (0.93 and 0.89, respectively). Notably, "Schizophrenia" and "Major depressive disorder" showed the lowest coherence not only in the "Psychiatric disease" category (0.67 and 0.75, respectively) but among all phenotypes. Other psychiatric disorders, such as "Bipolar disorder", "Autism", "ADHD", showed intermediate coherence estimates (0.87, 0.83, 0.78, respectively). In summary, these low coherence estimates highlight the need for further research on genetics of most cardiovascular and psychiatric diseases to better understand their molecular mechanisms, and suggests "Schizophrenia" and "Major depressive disorder" as the top candidates for further genetic studies.

# Discussion

We developed a general-purpose method to measure and compare coherence estimates of the molecular mechanisms in networks of genes. Intuitively, gene networks with high internal and low external interactions would be considered highly coherent; consequently, coherence is defined as the ratio between the internal and external degree distributions (the number of interactions per gene). We normalized coherence estimates to the range of high and random coherence, exemplified by KEGG pathways and random networks. To decouple coherence from its dependence on network size, we used KEGG and random networks matched in size to phenotype-specific networks. Applied to the analysis of 133 disease- and trait-associated gene sets, we found an unexpected inverse association of phenotype complexity with coherence. Consequently, complex traits like "Height", "Intelligence", "Longevity" had overall low coherence while metabolic traits and diseases had high coherence. Diseases in "Psychiatric" category had overall low coherence, with "Schizophrenia" being least coherent. Our method allows one to quantify the estimated coherence of molecular mechanisms across phenotypes, and will only improve as more PPI information becomes available.

Our approach is similar in spirit to the well-researched community detection problem [@danon2005comparing; @fortunato2010community; @Radicchi:2004aa; @Leskovec:2010:ECA:1772690.1772755]. However, we approach this problem from a different angle by asking whether a given network as a whole can be considered a community within the global PPI network. This may be considered a drawback as it has been shown that many networks consist of miniature communities known as motifs [@Milo:2002aa], or graphlets [@Przulj:2007aa]. While the investigation of motif enrichment [@Milo:2002aa] is a viable approach to characterize a network, our goal was to develop a unified measure of coherence that can be compared across phenotypes.

Our method relies on the ability to link the phenotype-associated genetic variants detected in genome-wide association studies to genes thought to be affected by them. Typically, genes are linked to genetic variants by proximity (nearest gene), a strategy adopted by the NHGRI-EBI GWAS catalog [@MacArthur:2017aa]. <!--Numerous tools have been created to associate genetic variants with genes (MAGENTA [@Segre:2010aa], VEGAS [@Liu:2010aa], SKAT [@Ionita-Laza:2013aa], to name a few, reviewed in [@Wang:2011ab]).--> However, the majority of phenotype-associated genetic variants are located in regulatory elements outside of protein-coding regions [@Khurana:2013aa]. <!--classical genetic studies of translocations and deletions have shown that distal regions, located at tens to hundreds of kilobases from a gene, affect transcription, indicating that regulatory elements can exert their effect over large genomic distances [@Kleinjan:2005aa].-->The pilot project by ENCODE showed that fewer than 10% of all interactions between TSSs and regulatory regions involved the closest TSS by linear distance [@Sanyal:2012aa], questioning the validity of the nearest gene mapping strategy. The development of chromatin conformation capture technologies now allows for understanding interactions among genomic regions on a genome-wide scale [@Lieberman-Aiden:2009aa; @Zhang:2013aa], helping to prioritize disease-associated gene-variant associations [@Pan:2018aa]. Consequently, tools are being developed to consider long-distance interactions when mapping SNPs to genes (3DSNP [@Lu:2017aa], FUMA [@Watanabe:2017aa], PINES [@Bodea2016], HUGIn [@Martin:2017aa]), and studies redefining disease-associated genes using long-distance interactions started to emerge [@Fang:2019aa]. Our future goal includes considering long-distance variant-gene interactions in defining phenotype-associated gene sets and comparing their network properties.

The definition of coherence as the ratio between internal vs. external degree distributions makes our measure dependent on the choice of PPI data [@Bajpai:2019aa]. Our limited knowledge of protein-protein interactions limits the coherence estimation to phenotypes that have a sufficient number of genes with PPI annotations, especially for phenotypes with low numbers of associated genes. Furthermore, networks formed by phenotype-associated genes should be sufficiently diverse to avoid cases in which all genes in a network have the same degree. As the different PPI databases vary significantly in the number of PPI annotations, some phenotype-associated gene sets have sufficient PPI annotations when using one but not the other database. This problem is best illustrated by the BioGrid PPI database that has considerably fewer PPIs; hence, fewer phenotypes could be analyzed. Differences in PPI databases may also lead to the presence of outliers that significantly alter coherence estimates. An example of such instability includes ESR2 and MOV10 genes in "Atrial Fibrilation" and "Blood Pressure", respectively, that have unusually high number of external edges in Biogrid- but not STRING-based networks. Consequently, the corresponding networks appear incoherent/highly coherent, when using Biogrid and STRING PPI databases, respectively. The differences in database-specific networks are alleviated by testing the significance of coherence using the permutation test; however, care should be exercised when selecting the PPI database for network analyses [@Bajpai:2019aa].

<!--The definition of high and random coherence also depends on the choice of PPI data. Therefore, we derived degree distributions for KEGG and random networks separately for each PPI database and implemented the max-min normalization of the coherence estimates. Intuitively, the max-min normalization converts the absolute measures of coherence (slopes of regression lines fit through the internal vs. external degree distribution plots) that are dependent on the choice of PPI databases to the percentages of the full range of coherence (from high coherence defined by KEGG networks to low coherence defined by random networks) defined using a given PPI database. The phenotype-specific coherences expressed as a percentage of the full range of coherence can be directly compared across phenotypes and databases.

Although the normalized coherence derived using different PPI databases can be compared among phenotypes, it tells nothing whether a given coherence estimate is significantly different from random. We developed a permutation test aimed to provide a measure of the stability of the coherence. Intuitively, the permutation test defines how different the coherence of a given phenotype from that of random networks of the same size. It provides additional protection against cases where coherence may appear high, yet, it may be a consequence of small network size providing unstable degree distributions. Testing the coherence estimates for significance helps against conclusions of the overall low coherence (e.g., anthropometric traits, Figure 3), when the coherence of individual traits is high (e.g., individual anthropometric traits, Figure 5B). Coupled with the permutation test, our coherence estimation provides a robust framework for quantifying the coherence estimates of phenotype-associated genes, advancing our understanding of the corresponding molecular mechanisms.-->

In addition to internal vs. external degree distributions, other network properties can be used to estimate coherence [@Leskovec:2010:ECA:1772690.1772755]. For example, the centrality measure has been used to demonstrate that highly central hub genes are functionally essential and highly conserved [@Khurana:2013aa]. A large study of PPIs (BioPlex) compared centrality distributions among several disease categories. Cancers and immunological disease networks tended to have high centrality, while nervous system, congenital, neonatal and hereditary disease networks had low centrality [@Huttlin:2017aa]. Metabolic pathways were shown to have more duplicated copies of highly central genes making the networks tolerant to loss-of-function mutations [@Khurana:2013aa]. In summary, centrality measure parallels our observations of network coherence. Our future work will incorporate centrality and other network metrics into our framework to further refine our ability to compare coherence of phenotype-associated molecular mechanisms.

[??? Potentially. discuss: We do not account for configuration]

## Conclusion

Our study investigated the coherence of the molecular mechanisms of genes associated with genomic variants of 133 phenotypes. We developed a network coherence measure that estimates the degree of internal vs. external connectivity that is robust to the difference in network sizes. Using well-curated lists of genomic variants and the associated genes, we built disease-specific networks using protein-protein interaction information from the STRING and BioGrid databases. We found a range of coherence estimates in each phenotype category, with metabolic diseases and traits being the most coherent. Psychiatric disorders were, on the other hand, the least coherent, with schizophrenia and major depressive disorder being the least coherent among all phenotypes. In summary, we provide a general-purpose method for quantifying and comparing coherence of molecular networks.

# Methods

## Data sources

Trait-associated gene lists were obtained from the NHGRI-EBI catalog of genome-wide association studies (GWAS catalog, `gwascat` v.2.14.0 R package) [@Hindorff:2009aa; @MacArthur:2017aa]. "MAPPED_GENE" column was used to extract trait-associated genes. To organize traits into categories, Experimental Factor Ontology (EFO) IDs were extracted from the "MAPPED_TRAIT_URI" column. Categories were mapped to EFO IDs using the `ontoCAT` v.1.12.0 R package and the EFO database (accessed 12/04/2018). In the case of a trait being mapped into two categories, the most representative category was manually assigned. Non-canonical gene names were converted to common gene symbols using `alias2Symbol` function from the `limma` R package. Gene names which could not be mapped to gene symbols were excluded from the analysis.

Human protein-protein interactions (PPIs) were obtained from STRING database v.11.0 [@Szklarczyk:2015aa] and BioGrid database v.3.5.174 [@Chatr-Aryamontri:2017aa] (accessed 06/25/2019). The STRING data for the tax ID "9606" (human) was used, and Ensembl protein IDs were converted to gene names using BioMart. Either the complete dataset (11,761,040  entries) or data filtered by "combined_score" > 500 (range 0-1000) (1,373,946 entries, the main PPI dataset) was used. The BioGrid data was filtered to include "TaxID interactor" type equal to "9606", all "Interaction Types" were included, totaling 465,660 interactions. These three PPI databases were used in parallel to evaluate disease coherence.

Trait-associated gene networks were formed by mapping trait-associated genes to PPIs. Genes without PPI annotations (e.g., C5orf4, LOC100507462), antisense transcripts (e.g., CDKN2B-AS), readthrough (e.g., NPHP3-ACAD11), non-protein-coding transcripts (e.g., LINC00478) were excluded. Genes that do not have PPIs with each other but interact with other genes in the global PPI network were kept. Traits with less than five genes were omitted. Additionally, traits with genes having identical internal degrees (hence, no degree distribution) were filtered due to the inability to derive a fit through the internal vs. external degree distributions (see below). These precautions were set to avoid situations where a small number of genes in a network can cause instability in the results. Importantly, the number of trait-associated genes having PPI annotations depends on the content of a PPI database; hence, some traits may be filtered when using one but not the other PPI database to build trait-associated networks.

## Comparing internal vs. external degree (aka connectivity) distributions as a measure of coherence

Trait-specific **internal degree distribution** was defined as the number of interactions between trait-associated genes, or, more precisely, their interacting protein products. As different traits have different numbers of associated genes forming networks of unknown configurations, the corresponding internal degree distribution was normalized to the expected interaction count for a network of that size (see Results). 

Trait-specific **external degree distribution** was defined as the number of interactions between genes in a trait-associated network and all other genes in the PPI network. Being measured in the context of the entire PPI network of fixed size, the external degree distribution did not require normalization. 

To compare trait-specific degree distributions, the normalized internal vs. external degree distributions were plotted on a scatterplot. The square root of the degree was taken for more informative representation. For each trait, the internal vs. external degree distribution plot was fit with a regression line through the origin to capture the slopes. Slopes were used as a measure of coherence with lower slopes being associated with higher coherence. Degree distributions of the network nodes were obtained using the `igraph` R package v. 1.0.1.

## Max-min normalization of coherence

The slopes of the internal vs. external degree distributions of the networks exhibiting high and random coherence (KEGG and random networks, respectively)  are dependent on the content of a PPI database. Thus, we normalized the slopes for the trait-associated networks using slopes formed by the degree distributions of similarly sized networks with high coherence ($\beta_{min}$ slope of KEGG networks) and random coherence ($\beta_{max}$ slope of random networks). For each trait $i$ with slope $\beta_i$, we calculate the normalized slope $\beta_{i\ norm} = \frac{\beta_{i} - \beta_{max}}{\beta_{min}- \beta_{max}}$, which gives a measure of coherence on the range of [0,1]. Higher values indicate a higher level of coherence similar to that of KEGG pathways. These values represent normalized coherence and can be compared across traits and PPI databases. 

## Permutation test of slopes

To test whether each slope is significantly different than random, we used a permutation test. For trait $i$ with the gene network of size $g_i$, we define $\beta_i$ as the slope derived by regressing the external degree distribution on the internal degree distribution. We then generate $n=10000$ networks of randomly selected genes of the same size $g_i$ as the set for trait $i$. For each random network $j$, we identify its slope $\beta_{r}$ representing the internal vs. external degree distribution regression line of random coherence. We calculate the permutation p-value as $p_i = \frac{(\sum_{j=1}^nI({\beta_{rj} > \beta_i}) + 1}{n + 1}$ [@Phipson:2010aa]. Intuitively, this approach estimates whether the slope of the trait-specific network $\beta_i$ is consistently larger than those of random networks. All analyses and visualizations were performed in the R/Bioconductor computing environment v.3.4.0 [@gentleman2006bioinformatics].

<!--For each disease, we find the absolute value of the difference between its slope and the random coherence slope. Next, we combine the genes for each disease with those of the random diseases and take two random samples of equal size to the disease gene set. We then calculate the absolute difference in slopes between the randomly sampled gene sets. This process is repeated 1000 times. Finally, we obtain a permutation p-value by taking the number of times the difference in slopes between random samples were less than the overall difference in slopes and dividing by 1000. -->

# Abbreviations

- GWAS - Genome-Wide Association Study
- KEGG - Kyoto Encyclopedia of Genes and Genomes
- PPI - Protein-Protein Interaction
- SNP - Single Nucleotide Polymorphism

# Acknowledgements

This work was supported by a Genetics and Human Agency Award from the Templeton Foundation to KSK, MR, CC, SB. The content is solely the responsibility of the authors and does not necessarily represent the official views of the Templeton Foundation. The authors would like to thank Spiro Stilianoudakis for help with Figure 1.

## Competing interests

The authors declare that they have no competing interests.

# Supplementary material

**Supplementary Table S1. Summary statistics of the analyzed phenotypes.** Minimum, mean, median, maximum of coherence estimates, the number of genes, SNPs, the total number of phenotype networks. Each worksheet corresponds to the PPI database used.

**Supplementary Table S2. Summary of networks from MSigDB categories.** Minimum, mean, median, maximum of the slopes, the number of genes, and the total number of networks are shown for each collection, along with the correlation between network size and slope. "CP:/KEGG/Reactome/Biocarta" - canonical pathways, "GOBP/GOMF/GOCC" - gene ontology biological processes, molecular functions, cellular component, "TFT" - transcription factor targets, "CGN" - cancer gene neighborhoods, "CM" - cancer modules, "MIR" - microRNA targets, "CGP" - chemical and genetic perturbations.

**Supplementary Table S3. Phenotype-specific coherence estimates.** The PPI database (STRING, STRING filtered, Biogrid) and the reference of high coherence (KEGG, GOCC, Reactome) are specified in the corresponding columns for coherence. Permutation p-values were estimated using STRING filtered database and KEGG reference of high coherence. "Trait" and "Disease" sheets correspond to subsets of the main table containing coherence estimates for traits/diseases, respectively, sorted by descending "String Filt Coherence KEGG" column.

**Supplementary Figure S1. Normalized coherence is unaffected by network size.** Red/blue gradient and numbers represent Pearson correlation coefficients for each pairwise comparison of the number of SNPs, genes, and the normalized coherence estimates using the corresponding PPI databases and KEGG pathways as a reference of high coherence.

**Supplementary Figure S2. Coherence estimates are similar when using different references of high coherences and PPI databases.** Red/blue gradient and numbers represent Pearson correlation coefficients for each pairwise comparison of the normalized coherence estimates using the corresponding PPI databases and KEGG/Reactome pathways and Gene Ontology Cellular Component networks as a reference of high coherence.

**Supplementary Figure S3. Coherence estimates of traits using KEGG as a reference of high coherence.** Size of dots represents the level of normalized coherence (X-axis) for individual traits (Y-axis). Plots are faceted by PPI database used to estimate coherence. Missing entries indicate that, for a given PPI database, a phenotype-associated network could not be built.

**Supplementary Figure S4. Coherence estimates of traits using GOCC as a reference of high coherence.** See legend for Supplementary Figure S3.

**Supplementary Figure S5. Coherence estimates of traits using Reactome as a reference of high coherence.** See legend for Supplementary Figure S3.

**Supplementary Figure S6. Coherence estimates of diseases using KEGG as a reference of high coherence.** See legend for Supplementary Figure S3.

**Supplementary Figure S7. Coherence estimates of diseases using GOCC as a reference of high coherence.** See legend for Supplementary Figure S3.

**Supplementary Figure S8. Coherence estimates of diseases using Reactome as a reference of high coherence.** See legend for Supplementary Figure S3.

# References

<!--
## Single networks, or networks of modules?

Our study considers a network of disease-associated genes in its entirety. However, networks may be comprised of interacting modules, i.e., relatively isolated subnetworks [@Hartwell:1999aa]. Connectivity of such modules would be different from the overall network connectivity [@Ghiassian:2015aa]. Although numerous methods exist for module detection in a network [@Chen:2008aa; @Dobrin:2009aa; @Jia:2011aa; @Agrawal:2018aa], comparing modules between diseases is plagued by normalization issues described in the Results section Intuitively, large networks will have a higher chance to have subnetworks (modules), and vice versa. It remains to be determined whether disease-associated genes form modular networks and whether the modular structure differs between (categories of) diseases.

## Tissue specificity

The effect of disease-associated SNPs should be considered in cell-type-specific context [@Trynka:2013aa; @Dozmorov:2016aa; @Sanyal:2012aa]. The difference between neurological disease- and schizophrenia-associated SNPs was determined on the cell-type-specific level, with schizophrenia- but not neurological disease-associated SNPs being enriched in genes expressed in medium spiny neurons, pyramidal CA1, pyramidal SS and cortical interneurons [@Skene:2018aa]. While cell-type-specific gene expression profiles have been generated, protein-protein interactions remain generic and highly incomplete [@Menche:2015aa]. Even the latest efforts to compile PPIs from multiple data sources (InWeb InBioMap [@Li:2017aa]) or single-technology large-scale PPI profiling (BioPlex 2.0 [@Huttlin:2017aa]) generate collections of  non-cell-type-specific interactions. Future efforts in cell-type-specific PPI profiling are expected to boost research on the comparison of disease networks built from PPIs.

## What constitutes random networks?

The degree was found to follow a power-law [@Yook:2005aa] or a truncated power-law distribution [@Huang:2007aa].

## GWAS p-value, cutoff, sample size, harmonization issues

In GWAS, it is often assumed that local correlations between adjacent hypotheses along the genome are so strong that testing the whole genome is equivalent to performing only $10^6$ tests, regardless of the number of hypotheses tested. Under this assumption, FWER can be controlled using a fixed genome-wide significance level of $\alpha = 5*10^{-8}$, corresponding to a Bonferroni adjustment for $10^6$ tests. 
-->