# Coherence of Molecular Mechanisms in Genes Identified Through GWAS In Major Human Disease and Traits

Scripts to reproduce the analysis. Originally developed in https://github.com/cresswellkg/disease_coherence

Download the data 
```
cd Analysis/data
chmod +x get_data.sh
./get_data.sh
cd ..
```

In RStudio, set working directory to the `Analysis` folder (`setwd("Analysis")`) and run the scripts in the following order:

- `01_GWAS_Prep.Rmd` - Code for parsing through the ebicat and KEGG databases and creating gene lists for each disease and pathway. Creates `data/Disease_Genes` folder. 

- `02_Connectivity_Comparison.Rmd` - Calculation of degree distributions for disease related data. Results are produced for Biogrid and STRING. 
    - Creates `biogrid_edges_new.rds`, `string_edges_new.rds`, `string_filt_edges_new.rds`.

- `03_Normalized_Coherence.Rmd` - Scripts for calculating normalized coherence for phenotype-specific networks. 
    - Uses PPI data from Biogrid, String, String filtered and `*_edges_new.rds`, calculates slopes for MSigDb networks (KEGG, GOCC, Reactome) and saves them in `Biogrid_Msig_Slopes.rds`, `String_Msig_Slopes.rds`, `String_Filt_Msig_Slopes.rds`. 
    - Calculates normalized coherence for phenotype-specific networks and saves them in `data/CoherenceResults/coherence_biogrid.rds`, `coherence_string.rds`, `coherence_string_filt.rds`. 
    - Produces two additional files `data/CoherenceResults/All_Msig_String.rds` containing slopes for all Msigdf pathways and `data/CoherenceResults/string_random_edges.rds` containing slopes for randomly generated pathways.

-`04_Coherence_Calculations.Rmd` - Takes normalized coherence, p-values and SNP/Gene counts and puts them into tables. 
    - Uses `data/CoherenceResults/coherence_*.rds` files, `supplementary_table_diseases_selected.csv`, and files in the `Analysis/Permutation_Results` folder. Creates `./manuscript/tables/supplementary_table_S3.csv`.

-`05_Category_Tables.Rmd` - Produces tables and plots summarizing coherence of Msigdf pathways and diseases separated by coherence.  Relies on `./manuscript/tables/supplementary_table_S3.csv` created by `04_Coherence_Calculations.Rmd`. Creates `./manuscript/tables/Table_1.csv`, `./manuscript/tables/supplementary_table_S1.xlsx`, `supplementary_table_S2.xlsx`, `Fig2_Size_Slope_Plot.png`.

- `06_Analysis_Tables.Rmd` - Produces main and supplementary figures. Relies on `./manuscript/tables/supplementary_table_S3.csv`. Creates `./manuscript/Figures/*`

Files:

- `biogrid_edges_new.rds`, `string_edges_new.rds`, `string_filt_edges_new.rds` - generated by `02_Connectivity_Comparison.Rmd`

- `Biogrid_Msig_Slopes.rds`, `String_Filt_Msig_Slopes.rds`, `String_Msig_Slopes.rds` - generated by `03_Normalized_Coherence.Rmd`

## `data`

Contains the data necessary to run the scripts in the folder

- `get_data.sh` - Shell script for downloading Biogrid and STRING PPI data. Creates:
    - `9606.protein.links.v11.0.txt.gz` - STRING PPI database data
    - `BIOGRID-ALL-3.5.174.mitab.txt.gz` - Biogrid PPI database data

- `supplementary_table_diseases_selected.csv` - categorized phenotypes from GWAScatalog

### `Coherence_Results`

Contains data frames with normalized coherence and gene counts for individual diseases (`coherence_*.rds`), all MSigDB pathways (`All_Msig_String.rds`) and random networks of equal sizes to KEGG pathways (`string_random_edges.rds`). Generated by `03_Normalized_Coherence.Rmd`. 

### `Disease_Genes`

Contains individual files for all diseases in `data(ebicat37)` from the `gwascat` R package. Files are named according to the corresponding disease gene and each row is a gene symbol. Generated by `01_GWAS_Prep.Rmd` . Only diseases in `supplementary_table_diseases_selected.csv` are actually used in the analysis. 

### `Cluster_Scripts`

Permutation analysis scripts ran on the cluster. `biogrid_perm*.R`, `string_perm_cluster*.R`, `string_perm_filt*R` run the permutation analysis in batches using Biogrid, STRING, STRING filtered PPI databases, respectively. `bind*.R` combine the results to produce `Complete_10k_Biogrid.rds`, `Complete_10k_String.rds`, `Complete_10k_String_Filt.rds` files, to be placed in the `Permutation_Results` folder

### `Permutation_Results`

Due to long run times (multiple days) the scripts from `Cluster_Scripts` folder were run on the Linux cluster and the `*.rds` files created by them are incuded. Files are used by `04_Coherence_Calculations.Rmd`.


## `functions`

Contains functions for conversion of gene networks to degree distributions
